<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <meta name="generator" content="HTML Tidy for HTML5 for Linux version 5.7.16" />
  <title>Modeling Individual Cyclic Variation in Human Behavior</title><!-- Copyright (c) 2010-2015 The MathJax Consortium -->
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta name="viewport" content="width=device-width; initial-scale=1.0;" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <link media="screen, print" rel="stylesheet" href="../../../data/dl.acm.org/pubs/lib/css/bootstrap.min.css" />
  <link media="screen, print" rel="stylesheet" href="../../../data/dl.acm.org/pubs/lib/css/bootstrap-theme.min.css" />
  <link media="screen, print" rel="stylesheet" href="../../../data/dl.acm.org/pubs/lib/css/main.css" />
  <script src="../../../data/dl.acm.org/pubs/lib/js/jquery.min.js" type="text/javascript"></script>
  <script src="../../../data/dl.acm.org/pubs/lib/js/bootstrap.min.js" type="text/javascript"></script>
  <script src="../../../data/dl.acm.org/pubs/lib/js/bibCit.js" type="text/javascript"></script>
  <script src="../../../data/dl.acm.org/pubs/lib/js/divTab.js" type="text/javascript"></script>
  <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS_CHTML"></script>
  <script type="text/x-mathjax-config">
  <![CDATA[
  MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}});
  ]]>
  </script>
<link rel="cite-as" href="https://doi.org/10.1145/3178876.3186052"/></head>
<body id="main">
<div>
<p style='font-size: 75%; color #444'>
This is a web copy of <a href='https://doi.org/10.1145/3178876.3186052'>https://doi.org/10.1145/3178876.3186052</a>.
 Published in WWW2018 Proceedings © 2018 International World Wide Web Conference Committee, published under 
 <a rel='license' property='license' href='https://creativecommons.org/licenses/by/4.0/'>
 Creative Commons CC By 4.0 License</a>.
The <a href='https://github.com/usable-oa/thewebconf2018/tree/master/scripts'>modifications</a> 
from the original are solely to improve HTML aiming to make it Findable, Accessible, Interoperable and Reusable. 
augmenting HTML metadata and avoiding ACM trademark.
To reference this HTML version, use:
</p><p>
<strong>Permalink:</strong>
<a href='https://w3id.org/oa/10.1145/3178876.3186052'>https://w3id.org/oa/10.1145/3178876.3186052</a>
</p></div>
<hr>


  <section class="front-matter">
    <section>
      <header class="title-info">
        <div class="journal-title">
          <h1><span class="title">Modeling Individual Cyclic Variation in Human Behavior</span><br />
          <span class="subTitle"></span></h1>
        </div>
      </header>
      <div class="authorGroup">
        <div class="author">
          <span class="givenName">Emma</span> <span class="surName">Pierson</span>, Stanford University, <a href="mailto:emmap1@cs.stanford.edu">emmap1@cs.stanford.edu</a>
        </div>
        <div class="author">
          <span class="givenName">Tim</span> <span class="surName">Althoff</span>, Stanford University, <a href="mailto:althoff@cs.stanford.edu">althoff@cs.stanford.edu</a>
        </div>
        <div class="author">
          <span class="givenName">Jure</span> <span class="surName">Leskovec</span>, Stanford University, <a href="mailto:jure@cs.stanford.edu">jure@cs.stanford.edu</a>
        </div>
      </div><br />
      <div class="pubInfo">
        <p>DOI: <a href="https://doi.org/10.1145/3178876.3186052" target="_blank">https://doi.org/10.1145/3178876.3186052</a><br />
        WWW '18: <a href="https://doi.org/10.1145/3178876" target="_blank">Proceedings of The Web Conference 2018</a>, Lyon, France, April 2018</p>
      </div>
      <div class="abstract">
        <p><small>Cycles are fundamental to human health and behavior. Examples include mood cycles, circadian rhythms, and the menstrual cycle. However, modeling cycles in time series data is challenging because in most cases the cycles are not labeled or directly observed and need to be inferred from multidimensional measurements taken over time. Here, we present <em>Cyclic Hidden Markov Models</em> (CyHMMs) for detecting and modeling cycles in a collection of multidimensional heterogeneous time series data. In contrast to previous cycle modeling methods, CyHMMs deal with a number of challenges encountered in modeling real-world cycles: they can model multivariate data with both discrete and continuous dimensions; they explicitly model and are robust to missing data; and they can share information across individuals to accommodate variation both within and between individual time series.</small></p>
        <p><small>Experiments on synthetic and real-world health-tracking data demonstrate that CyHMMs infer cycle lengths more accurately than existing methods, with 58% lower error on simulated data and 63% lower error on real-world data compared to the best-performing baseline. CyHMMs can also perform functions which baselines cannot: they can model the progression of individual features/symptoms over the course of the cycle, identify the most variable features, and cluster individual time series into groups with distinct characteristics. Applying CyHMMs to two real-world health-tracking datasets—of human menstrual cycle symptoms and physical activity tracking data—yields important insights including which symptoms to expect at each point during the cycle. We also find that people fall into several groups with distinct cycle patterns, and that these groups differ along dimensions not provided to the model. For example, by modeling missing data in the menstrual cycles dataset, we are able to discover a medically relevant group of birth control users even though information on birth control is not given to the model.</small></p>
      </div>
      <div class="classifications">
        <div class="AcmReferenceFormat">
          <p><small><span style="font-weight:bold;">ACM Reference Format:</span><br />
          Emma Pierson, Tim Althoff, and Jure Leskovec. 2018. Modeling Individual Cyclic Variation in Human Behavior. In <em>WWW 2018: The 2018 Web Conference,</em> <em>April 23–27, 2018,</em> <em>Lyon, France. ACM, New York, NY, USA</em> 10 Pages. <a href="https://doi.org/10.1145/3178876.3186052" class="link-inline force-break" target="_blank">https://doi.org/10.1145/3178876.3186052</a></small></p>
        </div>
      </div>
    </section>
  </section>
  <section class="body">
    <section id="sec-4">
      <header>
        <div class="title-info">
          <h2><span class="section-number">1</span> Introduction</h2>
        </div>
      </header>
      <p>Detecting and modeling cyclic patterns is important for understanding human health and behavior [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0027">27</a>]. Cycles appear in domains as varied as psychology (<em>e.g.</em>, mood cycles [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0026">26</a>] and cyclic mood disorders [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0022">22</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0049">49</a>]); physiology (<em>e.g.</em>, circadian rhythms and hormonal cycles [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0001">1</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0014">14</a>]); and biology (<em>e.g.</em>, the cell cycle [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0042">42</a>]). Modeling such cycles enables beneficial health interventions: modeling mood cycles can predict symptoms and guide treatments [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0041">41</a>], and modeling circadian cycles can predict fatigue and improve safety [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0001">1</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0018">18</a>].</p>
      <p>In the cycles described above, a latent state progresses cyclically and influences observed data. Consider for concreteness the running example of hormone cycles. A person's hormonal state—that is, the levels of hormones in their bloodstream—is generally unobserved, but it progresses cyclically and can influence observed behavior&nbsp;[<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0050">50</a>]. For example, in women, a single hormone cycle consists of moving through a series of states—follicular, late follicular, midluteal, late luteal [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0007">7</a>]—before returning to the follicular state. (Throughout this paper, completing a <em>cycle</em> means progressing through all latent states in a defined order before returning to the original latent state.) Modeling hormone cycles includes several problems of interest in human health: How does one compute the length of a cycle for a particular person? Can we cluster individuals into groups with distinct cycle dynamics, who may benefit from different medical interventions? How do features/symptoms vary throughout the cycle, since features that increase at the same point in the cycle may have the same physiological cause? (We use <em>feature</em> to refer to a single observed dimension of a multivariate time series: in the hormone cycles case, a binary feature might be whether the person reported negative mood on each day, and a second feature might be whether they reported pain).</p>
      <p>Modeling cycles presents a number of challenges. The cycle state for each individual at each timestep is generally unobserved: a person's cyclic hormonal state may influence the level of positive affect they express on social media, but social media data is rarely linked to data on hormone state. In addition, cyclic dynamics vary across individuals—for example, individuals may experience cycles of slightly different lengths—and vary over time within a single individual as well. Further, real-world time series data is multivariate with discrete as well as continuous features and missing data.</p>
      <p>Numerous methods have been developed to model cyclic patterns, including Fourier transforms, autocorrelation, data mining techniques, and computational biology algorithms &nbsp;[<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0005">5</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0009">9</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0013">13</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0015">15</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0027">27</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0040">40</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0051">51</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0063">63</a>]. However, these techniques fail to address one or more of the challenges described above and, because they are not generative models, lack the ability to fully model the cycle: for example, they are not designed to model how each feature varies throughout the cycle or cluster individuals into distinct groups. In addition, previous work has been limited by the lack of ground-truth data on the true latent cycle state, which has prevented quantitative evaluation and comparisons between methods. That is, if an unsupervised method claims that cycles have a certain length or a certain feature or symptom varies particularly dramatically over the course of the cycle, how do we assess whether those inferences are accurate?</p>
      <p><strong>This work.</strong> Here we present a new method, <em>Cyclic Hidden Markov Models (CyHMMs)</em>, which models cycles and addresses the challenges described above. CyHMMs take as input a collection of multivariate time series, with one time series for every individual in a population; time series can have missing data and discrete or continuous values. For each individual and each timestep, CyHMMs infer a discrete latent cycle state from the observed time series, using the fact that features vary dynamically based on the latent cycle state. Given these inferred cycle states, CyHMMs can then (1) compute the <em>cycle length</em> as the time it takes an individual to return to the same latent state, (2) predict <em>feature trajectories</em> by using the fitted generative model to predict how each feature will change over the course of the cycle, (3) identify the <em>most variable features</em>, and (4) <em>cluster</em> individuals into groups with distinct cycle dynamics.</p>
      <p>We evaluate CyHMMs on both simulated data and two real-world cyclic datasets—of human menstrual cycle data, and activity tracking data—where we have knowledge of the true cycle state. Using simulated data with known parameters, we show that CyHMMs recover the cycle length 58% more accurately than the best performing baseline. On real-world data, CyHMMs recover the cycle length 63% more accurately than the best performing baseline. Furthermore, we show that CyHMMs can accurately infer other fundamental aspects of the cycle which existing methods cannot:</p>
      <ul class="list-no-style">
        <li id="list1" label="•"><em>Feature trajectories:</em> CyHMMs can infer how each feature or symptom fluctuates over the course of the cycle. This is important for understanding an individual's likely time course through the cycle, and for understanding which features tend to peak during the same cycle state, potentially indicating a common cause.<br /></li>
        <li id="list2" label="•"><em>Feature variability:</em> CyHMMs can infer which features are most variable. This is important for understanding which features are affected by the cycle, and which remain relatively constant.<br /></li>
        <li id="list3" label="•"><em>Clustering:</em> CyHMMs can partition individuals into groups which correlate with true population heterogeneity. This is important because not all individuals have the same cycle dynamics, and so different individuals may benefit from different interventions.<br /></li>
      </ul>
      <p>Our inferences on two real-world datasets further demonstrate the utility of cycle modeling in the important human health domains of activity tracking, where we show we can accurately recover weekly sleep cycles, and menstrual cycle tracking, where we show we can accurately recover the 28-day cycle. For example, in the menstrual cycle data, CyHMMs identify a subpopulation of users who are much more likely to be taking hormonal birth control—even though no information on birth control is provided to the model. Identifying such subpopulations is essential for accurately characterizing cycles (since women taking birth control experience different menstrual cycle dynamics [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0045">45</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0048">48</a>]) and to designing medical treatments (since women taking birth control require different pharmacologic interventions [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0030">30</a>] and have different risks of hormone-related cancers [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0044">44</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0046">46</a>]). We emphasize that while in this paper we analyze datasets where cycle states are known in order to prove that CyHMMs work, CyHMMs are generally applicable to time series where cyclic dynamics are suspected but cycle states are not known. Our CyHMM implementation is publicly available.</p>
    </section>
    <section id="sec-5">
      <header>
        <div class="title-info">
          <h2><span class="section-number">2</span> Related Work</h2>
        </div>
      </header>
      <p>Our work draws on an extensive literature of methods developed for detecting and modeling cyclic patterns; we briefly describe major approaches. <em>Fourier transforms</em> [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0009">9</a>] express a time-dependent signal as a sum of sinusoids or complex exponentials; the most significant frequencies in the signal can be extracted from the amplitudes of the Fourier coefficients. While Fourier transforms are a classic technique for univariate time series, they are not easily adapted to data which is multivariate, missing, or discrete. Further, because they are not generative models, they cannot easily be used to predict how features will vary over the course of the cycle. It is also not clear how to apply them to a population of individuals whose time series are different but related. Fourier transforms also assume that the cycle length in each signal remains constant, which is not true for individuals whose cycle lengths vary from cycle to cycle. (Wavelet transforms [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0015">15</a>] overcome this last deficiency but not the other ones.) Another common technique for detecting periodicity, <em>autocorrelation</em>, computes the correlation between a time series and a lagged copy of itself [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0051">51</a>]. It is mathematically related to the Fourier transform and suffers from similar drawbacks.</p>
      <p>More recent <em>data mining techniques</em> find “partial periodic patterns” that occur repeatedly in time series, sometimes with imperfect regularity [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0005">5</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0006">6</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0012">12</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0025">25</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0027">27</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0028">28</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0029">29</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0032">32</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0034">34</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0039">39</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0040">40</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0061">61</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0062">62</a>]. These methods represent discrete time series as strings and search for recurring patterns. Han et al. [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0027">27</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0028">28</a>] introduce the concept of partial periodicity; numerous related approaches have been developed, including Ma and Hellerstein's algorithm for detecting periodic events with unknown periods&nbsp;[<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0040">40</a>] and methods for detecting frequently recurring patterns&nbsp;[<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0025">25</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0029">29</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0047">47</a>]. These models do not address several challenges addressed in this work. They are not full generative models and do not recover all the cycle parameters in which we are interested; they cannot, for example, predict how observed features will vary over the course of the cycle. Second, we seek a method applicable to <em>both</em> discrete and continuous data, and string-based methods are inherently discrete.</p>
      <p>There is also a relevant literature in computational biology which searches for circadian rhythms, often in genomic data [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0016">16</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0033">33</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0060">60</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0063">63</a>], by fitting a periodic model to individual genes. These models have several drawbacks for the real-world datasets we consider: they are designed for data which is much higher-dimensional than most human activity datasets; they are designed for continuous data, and may not work well on discrete data; they consider each feature individually rather than sharing information across features; and they do not easily allow sharing of information across individuals.</p>
      <p>Because our method infers a hidden state at each timestep, it is also inspired by previous approaches which use latent states—and hidden Markov models specifically—to model time series. Such latent state approaches have been applied to model time series in domains such as speech recognition [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0052">52</a>], healthcare [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0058">58</a>] and sports [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0035">35</a>]. These approaches have been successful because multivariate time series in many domains are often generated by low-dimensional hidden states, motivating our application to cyclic time series. Our work also draws on ideas from hidden semi-Markov models (HSMMs) and explicit duration hidden Markov models (EDHMMs) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0021">21</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0037">37</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0065">65</a>] which develop formalisms for modeling latent states with non-geometric duration distributions that are commonly found in real-world data&nbsp;[<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0001">1</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0036">36</a>].</p>
    </section>
    <section id="sec-6">
      <header>
        <div class="title-info">
          <h2><span class="section-number">3</span> Task Description</h2>
        </div>
      </header>
      <p>We assume the data consist of <em>N</em> multivariate time series, one for each individual <em>i</em> in a population. Throughout, we use <em>feature</em> to refer to a single dimension of a multivariate time series. Each time series, <em>X</em> <sup>(<em>i</em>)</sup>, is a <em>T</em> <sup>(<em>i</em>)</sup> × <em>K</em> matrix, where <em>T</em> <sup>(<em>i</em>)</sup> is the number of timesteps in the time series and <em>K</em> is the number of features. <span class="inline-equation"><span class="tex">$X^{(i)}_{tk}$</span></span> denotes the value in the <em>i</em>-th time series at the <em>t</em>-th timestep of the <em>k</em>-th feature. Features may be binary or continuous and may have missing data. Our task is to infer basic cycle characteristics:</p>
      <ul class="list-no-style">
        <li id="list4" label="•"><em>Cycle length:</em> both for each individual and the whole population.<br /></li>
        <li id="list5" label="•"><em>Feature progression:</em> how each feature varies over the course of the cycle.<br /></li>
        <li id="list6" label="•"><em>Feature variability:</em> which features vary the most over the course of the cycle and which remain relatively constant.<br /></li>
        <li id="list7" label="•"><em>Clustering:</em> whether the population can be divided into groups with distinct cycle characteristics.<br /></li>
      </ul>
    </section>
    <section id="sec-7">
      <header>
        <div class="title-info">
          <h2><span class="section-number">4</span> Proposed Model</h2>
        </div>
      </header>
      <p>We propose a new class of generative models for cyclical time series data: <em>Cyclic Hidden Markov Model</em>s (CyHMMs). As in a standard HMM, at each timestep an individual is in one of <em>J</em> latent states <em>Z<sub>j</sub></em> , indexed by <em>j</em>, and emits an observed feature vector drawn from a distribution specific to that latent state. However, CyHMMs differ from standard HMMs in two ways. First, they capture cyclicity by placing constraints on how individuals can transition between states: each individual must progress through states <em>Z</em> <sub>1</sub>, <em>Z</em> <sub>2</sub>, ... in the same order (although they can begin in any start state) and after reaching the final state <em>Z<sub>J</sub></em> must return to <em>Z</em> <sub>1</sub> to begin the cycle again. An individual thus completes a <em>cycle</em> when they pass through all latent states and return to the same latent state. We constrain latent state progression in this way because it naturally encodes how the latent state in real-world cycles progresses cyclically and because it makes the model interpretable. If the latent state transitions were unconstrained it would not be clear how to define cycles.</p>
      <p>The second difference between our model and an ordinary HMM is that, when an individual enters a new state <em>Z<sub>j</sub></em> , they also draw a number of timesteps—a <em>duration</em>—to remain in that state before transitioning to the next state, an idea from hidden semi-Markov models [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0065">65</a>] (HSMMs) and explicit duration hidden Markov models [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0021">21</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0037">37</a>] (EDHMMs). In ordinary HMMs, the time spent in a state is drawn from a geometric distribution [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0017">17</a>], but as we discuss below, this parameterization is a poor approximation when modeling many real-world cycles; drawing state durations from arbitrary duration distributions allows for more flexible and realistic cycle modeling. Figure <a class="fig" href="#fig1">1</a> illustrates a path an individual might take through hidden states over the course of a single cycle. Each state <em>Z<sub>j</sub></em> is split into substates indexed by <em>d</em>, where <em>d</em> indicates the amount of time remaining in that state. Substate <em>Z<sub>jd</sub></em> progresses deterministically to <em>Z</em> <sub><em>j</em>(<em>d</em> − 1)</sub> unless <em>d</em> = 0, in which case it can progress to any substate of state <em>j</em> + 1 with probabilities drawn from <span class="inline-equation"><span class="tex">$\mathcal {D}_{j + 1}$</span></span> , where <span class="inline-equation"><span class="tex">$\mathcal {D}$</span></span> is the duration distribution. (If <em>j</em> = <em>J</em>, the model progresses instead to <em>Z</em> <sub>1</sub>, completing the cycle). All substates of state <em>Z<sub>j</sub></em> share the same emission distribution <span class="inline-equation"><span class="tex">$\mathcal {E}_j$</span></span> , and all transitions are either deterministic or parameterized by the duration distributions <span class="inline-equation"><span class="tex">$\mathcal {D}_j$</span></span> , so the model's emissions and transitions are completely parameterized by <span class="inline-equation"><span class="tex">$\mathcal {E}_j$</span></span> and <span class="inline-equation"><span class="tex">$\mathcal {D}_j$</span></span> <a class="fn" href="#fn1" id="foot-fn1"><sup>1</sup></a>. Expressing the model in this way is a commonly used computational technique&nbsp;[<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0065">65</a>], facilitating fast parameter fitting through optimized HMM packages [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0054">54</a>].</p>
      <figure id="fig1">
        <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186052/images/www2018-61-fig1.jpg" class="img-responsive" alt="Figure 1" longdesc="" />
        <div class="figure-caption">
          <span class="figure-number">Figure 1:</span> <span class="figure-title">Latent states in a CyHMM with J = 4 states and maximum state duration <em>d<sub>max</sub></em> = 2. The red circles and arrows illustrate one possible path through the CyHMM over the course of a single cycle. The <em>j</em>-th latent state counts down until the remaining duration in the state, <em>d</em>, is zero and then transitions to the next latent state. All substates with equal <em>j</em> have the same emission parameters.</span>
        </div>
      </figure>
      <p></p>
      <section id="sec-8">
        <header>
          <div class="title-info">
            <h3><span class="section-number">4.1</span> Model specification</h3>
          </div>
        </header>
        <p>Specifying a CyHMM requires specifying the emission distributions <span class="inline-equation"><span class="tex">$\mathcal {E}$</span></span> , the duration distributions <span class="inline-equation"><span class="tex">$\mathcal {D}$</span></span> , and the number of states <em>J</em>. We now describe and motivate our distributional choices. We find in our experiments that the distributions described below are sufficiently flexible to capture both simulated and real-world data, although our framework extends to other distributions.</p>
        <p><strong>Emission distributions.</strong> Human activity time series have frequent missing measurements caused, for example, by a user forgetting to track some event on a given day. We show in our real-world case studies (Section <a class="sec" href="#sec-15">6</a>), that the probability of missing data fluctuates over the course of the cycle and is important to model. Consequently, we use emission distributions which allow for missing data. We model the data using a mixture where each observation has some probability of being missing and is otherwise drawn from a Gaussian distribution (for continuous features) or a Bernoulli distribution (for binary features)<a class="fn" href="#fn2" id="foot-fn2"><sup>2</sup></a>.</p>
        <p><em>Continuous features</em>: if a user is in the <em>j</em>-th latent state at timestep <em>t</em>, we assume their value for the <em>k</em>-th feature, <em>X<sub>tk</sub></em> , is drawn as follows:</p>
        <div class="table-responsive">
          <div class="display-equation">
            <span class="tex mytex">\[ X_{tk} = \left\lbrace \begin{array}{ll}\rm {unobserved} &amp; \mbox{if } \text{Bernoulli}(p^\text{obs}_{jk}) = 0 \\ v_{tk} \sim \mathcal {N}(\mu _{jk}, \sigma _{jk}^2) &amp; \mbox{otherwise} \end{array} \right. \]</span><br />
          </div>
        </div>
        <p></p>
        <p>Hence, <em>X<sub>tk</sub></em> is unobserved with some probability <span class="inline-equation"><span class="tex">$1 - p^\text{obs}_{jk}$</span></span> and otherwise drawn from a Gaussian whose parameters are specific to that state and feature. The parameters for the <em>j</em>-th latent state and <em>k</em>-th feature are <em>μ<sub>jk</sub></em> , <em>σ<sub>jk</sub></em> , and <span class="inline-equation"><span class="tex">$p^\text{obs}_{jk}$</span></span> . Each feature is independent conditional on the hidden state: the emission probability for all features is the product of the emission probabilities for each feature.</p>
        <p><em>Binary features</em>: Binary features in real-world time series introduce a subtlety: if all features are zero, it is unclear whether they are true zeros or missing data. For example, if a person on a negative mood-tracking app records no negative mood symptoms, it is unclear whether they were content or whether they merely neglected to log their negative mood. (With continuous features, it is generally more obvious when data is missing, since zero values are out of range). Thus, we assume the following data-generating process: with some probability at each timestep, the person does not bother to log any features and data for all features is missing. Otherwise, they log at least one feature, and we assume they did not experience the features they did not log. Hence, if a person is in the <em>j</em>-th latent state at the <em>t</em>-th timestep, observed data is drawn as follows:</p>
        <div class="table-responsive">
          <div class="display-equation">
            <span class="tex mytex">\[ I^{\text{logged anything}}_t \sim \text{Bernoulli}(p^\text{obs}_j) \]</span><br />
          </div>
        </div>
        <div class="table-responsive">
          <div class="display-equation">
            <span class="tex mytex">\[ X_{tk} = \left\lbrace \begin{array}{ll}\rm {unobserved} &amp; \mbox{if } I^{\text{logged anything}}_t = 0 \\ v_{tk} \sim \text{Bernoulli}(E_{jk}) &amp; \mbox{otherwise} \end{array} \right. \]</span><br />
          </div>
        </div>The parameters of the emission distribution are <span class="inline-equation"><span class="tex">$p^\text{obs}_j$</span></span> and <em>E<sub>jk</sub></em> .
        <p></p>
        <p><strong>Duration distributions.</strong> For ordinary HMMs, the time spent in each state follows a geometric distribution. However, this parameterization does not describe many real-world cycles. For example, the length of the luteal phase in the human menstrual cycle is better described by a normal distribution [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0036">36</a>]. In particular, the monotonicity of the geometric probability mass function makes it a poor approximation to many real-world distributions. Instead, we use Poisson distributions for the duration distributions <span class="inline-equation"><span class="tex">$\mathcal {D}_j$</span></span> , so the duration distribution for the <em>j</em>-th latent state is parameterized by a single rate parameter <em>λ<sub>j</sub></em> . We use Poisson distributions because they are frequently used in HSMMs, they are fast to fit and easy to interpret, and we show empirically that they provide a better fit to real-world cycles than does an implementation with a geometric distribution. Our framework extends to other distributions.</p>
        <p><strong>Choosing the number of hidden states.</strong> <em>J</em> can be chosen either based on prior knowledge about the cycle being modeled (as we do in our case studies) or using cross-validated log likelihood [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0011">11</a>].</p>
      </section>
      <section id="sec-9">
        <header>
          <div class="title-info">
            <h3><span class="section-number">4.2</span> Model fitting</h3>
          </div>
        </header>
        <p>The optimization procedure is an expectation-maximization algorithm often used with HMMs; we outline it briefly.</p>
        <ul class="list-no-style">
          <li id="list8" label="•">
            <strong>E-step</strong>: Use the Baum-Welch algorithm [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0008">8</a>] to infer the probability of being in a given latent state at a given timestep, <span class="inline-equation"><span class="tex">$p_j^{(t)}$</span></span> , and the expected number of transitions between each pair of substates <em>Z<sub>jd</sub></em> and <span class="inline-equation"><span class="tex">$Z_{j^{\prime }d^{\prime }}$</span></span> . Because of the model structure (Figure&nbsp;<a class="fig" href="#fig1">1</a>), the expected number of transitions is trivial for all start substates with <em>d</em> ≠ 0 (since the substate must simply count down to the next substate). Denote by <span class="inline-equation"><span class="tex">$C^{(i)}_{jd}$</span></span> the expected number of transitions in time series <em>i</em> into substate <em>Z<sub>jd</sub></em> from the prior substate <em>Z</em> <sub>(<em>j</em> − 1)0</sub>.<br />
          </li>
          <li id="list9" label="•">
            <strong>M-step</strong>: Update the parameters of the duration distribution <span class="inline-equation"><span class="tex">$\mathcal {D}_j$</span></span> and the emission distribution <span class="inline-equation"><span class="tex">$\mathcal {E}_j$</span></span> for each latent state.<br />
            <ul class="list-no-style">
              <li id="list10" label="–">Updating <span class="inline-equation"><span class="tex">$\mathcal {D}_j$</span></span> : In the case of the Poisson distribution, each state has a single rate parameter <em>λ<sub>j</sub></em> which is estimated as the mean expected duration in that state:
                <div class="table-responsive">
                  <div class="display-equation">
                    <span class="tex mytex">\[ \lambda _j = \frac{\sum _{i = 1}^N\sum _{d = 1}^{D} C^{(i)}_{jd} \cdot d}{\sum _{i = 1}^N\sum _{d = 1}^{D} C^{(i)}_{jd}} \]</span><br />
                  </div>
                </div>We set the probability of beginning in each substate <em>Z<sub>jd</sub></em> to be <em>f<sub>j</sub></em> (<em>d</em>), where <em>f<sub>j</sub></em> is the duration probability mass function for the <em>j</em>-th latent state.<br />
              </li>
              <li id="list11" label="–">Updating <span class="inline-equation"><span class="tex">$\mathcal {E}_j$</span></span> . We update the emission distribution parameters by computing their sufficient statistics. For discrete distributions, the sufficient statistics are the sum of weights <span class="inline-equation"><span class="tex">$p_j^{(t)}$</span></span> over non-missing samples, the sum of weights over all samples, and the weighted sum of each feature over non-missing samples. For continuous distributions, we must also compute the weighted sum of the square of each feature over non-missing samples. The updates are straightforward; we omit them due to space constraints and provide our implementation online.<br /></li>
            </ul>
          </li>
        </ul>
        <p><strong>Violation of model assumptions.</strong> Model fitting will converge regardless of whether there are actually cycles in the data, although convergence may be slow if model assumptions are seriously violated. Fit should be assessed by comparing model parameters and the outputs discussed in Section <a class="sec" href="#sec-10">4.3</a> with prior knowledge about the cycle to determine the plausibility of the fitted model.</p>
        <p><strong>Parameter initialization.</strong> CyHMMs are initialized with a hypothesized cycle length. In our simulations (Section <a class="sec" href="#sec-11">5</a>), we found that CyHMMs are fairly robust to inaccurate specification of this parameter: they converged to the true length as long as they were initialized within roughly 50% of the true length. In real-world datasets, this is a reasonable constraint: the true cycle length is often at least approximately known (as it is in the two real-world case studies we consider in Section <a class="sec" href="#sec-15">6</a>). To further increase robustness, the model can be fit using a range of cycle length initializations, and the model with the best log likelihood can then be used; we confirm on simulated data that this works reliably.</p>
        <p><strong>Scalability.</strong> Three features allow our implementation to scale to real-world datasets: we specifically adapt our implementation to use an optimized HMM library [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0054">54</a>]; model fitting can be easily parallelized because the forward-backward computations take up most of the computation time and are independent for each time series; and we find that in real-world applications a relatively small number of hidden states suffices to capture realistic dynamics. Consequently, as we describe in Section <a class="sec" href="#sec-15">6</a>, we are able to fit a CyHMM model in 3 minutes (using a computer with 16 cores) on an activity dataset 8 million total observations; we are able to fit a menstrual cycles dataset with 22 million total observations in 18 minutes (Figure <a class="fig" href="#fig2">2</a>). (By “observation”, we here mean a reading of one feature for one individual at one timestep). Both these datasets were provided by companies with large user bases and we needed to perform no filtering for scalability; all filtering was performed only to select active users who could provide rich data. For extremely large datasets, a random sample of users could also be used to fit the model, since statistical power would not be a concern.</p>
        <figure id="fig2">
          <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186052/images/www2018-61-fig2.jpg" class="img-responsive" alt="Figure 2" longdesc="" />
          <div class="figure-caption">
            <span class="figure-number">Figure 2:</span> <span class="figure-title">Time to fit CyHMM models on our real-world datasets. To provide runtimes on a range of sample sizes, we take random subsamples of the dataset. The menstrual cycles dataset is larger, so we can extrapolate further.</span>
          </div>
        </figure>
        <p></p>
      </section>
      <section id="sec-10">
        <header>
          <div class="title-info">
            <h3><span class="section-number">4.3</span> Inference of cycle properties</h3>
          </div>
        </header>
        <p><strong>Cycle length.</strong> To infer cycle length for each individual, we fit a CyHMM to the entire population, then use the fitted model to infer the Viterbi path—that is, the most likely sequence of hidden states—for each individual from their observed data. We define the inferred cycle length as the time to return to the same hidden state in the Viterbi path.</p>
        <p><strong>Feature trajectories.</strong> We use CyHMMs to infer how features fluctuate (increase or decrease) over the course of a cycle as follows. After fitting the CyHMM, we assume all users are in a single substate and propagate the state vector forward by multiplying by the learned transition matrix. We compute the expected value of each feature at each timestep using the learned emission model.</p>
        <p><strong>Feature variabilities.</strong> We define feature variability to be how much a feature fluctuates over the course of a cycle and compute it as follows. Given the inferred trajectory for feature <em>k</em>—a value <em>V<sub>tk</sub></em> at each timestep <em>t</em> less than or equal to the cycle length <em>L</em>—we define the feature variability <em>Δ<sub>k</sub></em> as how much the feature varies relative to its mean <em>μ<sub>k</sub></em> : <span class="inline-equation"><span class="tex">$\Delta _{k} = \frac{1}{L}\sum _{t = 1}^{\text{L}} \big |\frac{V_{tk} - \mu _k}{\mu _k}\big |$</span></span> . (We divide by the mean to ensure features with different scales can be compared; this definition of <em>Δ<sub>k</sub></em> is similar to the coefficient of variation [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0010">10</a>].) Our definition of feature variability is not equivalent to merely computing the standard deviation of all observations of a feature: for example, if our feature is “person carries umbrella”, its variability over the menstrual cycle will be zero because rain (and umbrellas) do not vary with hormone state.</p>
        <p><strong>Clustering.</strong> Previous work has found that clustering time series can identify important heterogeneity [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0038">38</a>]; we therefore extend our model to divide individuals into clusters with similar cycle progression patterns. After initializing the clustering by dividing individuals into <em>C</em> clusters<a class="fn" href="#fn3" id="foot-fn3"><sup>3</sup></a>, we use an EM procedure with hard cluster assignment, alternating between two steps until convergence: 1) for each cluster <em>c</em>, fit a separate CyHMM <span class="inline-equation"><span class="tex">$\mathcal {M}_c$</span></span> using only individuals in that cluster; 2) reassign each individual <em>i</em> to the cluster <em>c</em> whose model maximizes the log likelihood of <em>X</em> <sup>(<em>i</em>)</sup>.</p>
      </section>
    </section>
    <section id="sec-11">
      <header>
        <div class="title-info">
          <h2><span class="section-number">5</span> Evaluation on Simulated Data</h2>
        </div>
      </header>
      <p>We first validate the performance of CyHMMs on synthetic data before examining their application to two real-world datasets.</p>
      <section id="sec-12">
        <header>
          <div class="title-info">
            <h3><span class="section-number">5.1</span> Synthetic data generation</h3>
          </div>
        </header>
        <p>We simulate a population where the data for each individual consists of a multivariate time series with missing data. We briefly describe the simulation procedure here and make the full implementation available online. To test our model's robustness to model misspecification, we do not generate data using the CyHMM generative model. In our simulation, each individual progresses through a cycle, and both the values for each feature and the probability that data for the feature is missing vary <em>sinusoidally</em> as a function of where the individual is in their cycle. We examine the effects of altering the maximum individual time series length (<em>T</em><sub>max</sub>), the amount of noise in the data (<em>σ<sub>n</sub></em> ), the probability data is missing (<em>p</em><sub>missing</sub>), and the between-user and within-user variation in cycle length (<em>σ<sub>b</sub></em> and <em>σ<sub>w</sub></em> ). We generate simulated data in two stages. First, we simulate the fraction of the way each person is through their cycle at each timestep, which we refer to as “cycle position”; cycle position can be thought of as a continuously varying cycle state. Second, we generate observed data given the person's cycle position.</p>
        <p><strong>Simulating cycle position.</strong> We draw each individual's average cycle length from <span class="inline-equation"><span class="tex">$L^{(i)} \sim N(L, \sigma _b^2)$</span></span> , where <em>L</em> is the average population cycle length and <em>σ<sub>b</sub></em> controls <em>between individual</em> variation in cycle length; we set <em>L</em> = 30 days in our simulations. We draw each individual's starting cycle day from a uniform distribution over possible cycle days; each subsequent day, their cycle day advances by 1, unless they have reached their current cycle length, in which case they go back to 0 and we draw the length for their next cycle from <span class="inline-equation"><span class="tex">$N(L^{(i)}, \sigma _w^2)$</span></span> . <em>σ<sub>w</sub></em> controls <em>within individual</em> variation in cycle length. We define an individual's <em>cycle position</em> <span class="inline-equation"><span class="tex">$\phi ^{(i)}_t \in [0, 1]$</span></span> to be the fraction of the way the individual is through their current cycle: <em>i.e.</em>, cycle day divided by current cycle length.</p>
        <p><strong>Simulating time series given cycle position.</strong> Given individual <em>i</em>’s cycle position at timepoint <em>t</em>, <span class="inline-equation"><span class="tex">$\phi ^{(i)}_t$</span></span> , we generate the values for a feature <em>k</em>, <span class="inline-equation"><span class="tex">$X_{tk}^{(i)}$</span></span> , as follows. We generate both binary and continuous data to evaluate our model under a wide variety of conditions. For continuous features, we allow the probability each feature is observed at each timestep, <span class="inline-equation"><span class="tex">$p_{tk}^{(i)}$</span></span> , to vary sinusoidally in <span class="inline-equation"><span class="tex">$\phi ^{(i)}_t$</span></span> . We draw whether the feature is observed from <span class="inline-equation"><span class="tex">$\text{Bernoulli}(p_{tk}^{(i)}$</span></span> ). If the feature is observed, we set its value to a second sinusoidal function of <span class="inline-equation"><span class="tex">$\phi ^{(i)}_t$</span></span> . For both sinusoids, we draw individual-specific coefficients to allow for variation in cycle dynamics across individuals. Our emission model for binary features is similar to the continuous model except that at each timestep, a single Bernoulli draw determines whether <em>all</em> features are missing, and we draw each feature from a Bernoulli whose emission probability varies sinusoidally in cycle position. Our full procedure for generating simulated data is available online.</p>
      </section>
      <section id="sec-13">
        <header>
          <div class="title-info">
            <h3><span class="section-number">5.2</span> Inference of cycle length</h3>
          </div>
        </header>
        <p>A basic question about any cycle is its length, both in individuals and in the population as a whole. We thus first evaluate how well CyHMMs infer the true cycle length for each individual as compared to baselines. Our evaluation metric is mean error.</p>
        <p><strong>Baselines.</strong> Based on the previous literature we describe in Section <a class="sec" href="#sec-5">2</a>, we compare to baselines from all three major areas of related work: classical techniques (Fourier transform and autocorrelation), data-mining techniques (Partial Periodicity Detection) and circadian rhythm detection (the ARSER algorithm):</p>
        <ul class="list-no-style">
          <li id="list12" label="•">
            <strong>Discrete Fourier transform</strong> [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0009">9</a>]: for each feature, we perform a discrete Fourier transform and take the period of the peak with the largest amplitude. We take the median of these periods across all features. (Fourier transforms are generally used on univariate data; by taking the median, we combine data from all features.)<br />
          </li>
          <li id="list13" label="•">
            <strong>Autocorrelation</strong> [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0051">51</a>]: for each feature, we compute the lag which produces the maximum autocorrelation in that feature. We take the median of these lags across all features.<br />
          </li>
          <li id="list14" label="•">
            <strong>Partial Periodicity Detection</strong> [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0040">40</a>] (binary data only): we apply Ma and Hellerstein's <em>χ</em> <sup>2</sup> test for partial periodicity detection<a class="fn" href="#fn4" id="foot-fn4"><sup>4</sup></a> to each individual feature. The algorithm returns a list of statistically significant periods; we take the median of these periods. The algorithm relies on specification of a <em>δ</em> parameter, which controls the time tolerance in period length; we evaluate <em>δ</em> = 2, 5, 10, find the algorithm yields the most accurate cycle length inference with <em>δ</em> = 2, and report results for this parameter setting.<br />
          </li>
          <li id="list15" label="•">
            <strong>ARSER circadian rhythm detection</strong> [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0063">63</a>]: the ARSER algorithm fits a sinusoidal model to each individual feature<a class="fn" href="#fn5" id="foot-fn5"><sup>5</sup></a>, and reports a <em>p</em>-value for periodicity. Consistent with the original authors, we filter for features which show statistically significant periodicities after multiple hypothesis correction; we then take the period with the largest amplitude for each feature and aggregate across multiple features by taking the median. We find that setting a statistical significance threshold of <em>p</em> = .01 yields the most accurate cycle length inference, and report results for this parameter setting. We note that the original implementation does not scale to datasets of our size, so to evaluate it we must make two modifications: we parallelize it to use multiple cores, and we fit the autoregression models using only the Yule-Walker and Burg methods, because the MLE method does not scale.<br />
          </li>
        </ul>
        <p>In assessment, we filter out periods which are shorter than 5 days or longer than 50 days (the true cycle length is 30 days). This increases accuracy by, for example, preventing the autocorrelation baseline from returning implausibly short lags (which in real-world settings an analyst would filter out) which degrade performance. We run 100 binary simulation trials and 100 continuous simulation trials, systematically varying the maximum individual time series length (<em>T</em><sub>max</sub>) from 90 to 180 days, the amount of noise in the data (<em>σ<sub>n</sub></em> ) from 5 to 50, the probability data is missing (<em>p</em><sub>missing</sub>) from 0% to 90%, and the between-user and within-user variation in cycle length (<em>σ<sub>b</sub></em> and <em>σ<sub>w</sub></em> ) from 1 to 10 days.</p>
        <p><strong>Results.</strong> CyHMMs have lower mean error than do baselines on both binary and continuous data (Table <a class="tbl" href="#tab1">1</a>). Across all 200 trials, they reduce the mean error in cycle length inference over the baseline with lowest mean error, autocorrelation, by 58%. Their inferred cycle lengths for each person are also better correlated with the true cycle lengths for each person (<em>r</em> = .57) than the baseline with the best correlation (ARSER, <em>r</em> = .19) demonstrating that they can better identify individual heterogeneity.</p>
        <div class="table-responsive" id="tab1">
          <div class="table-caption">
            <span class="table-number">Table 1:</span> <span class="table-title">Mean error in inferring cycle length across simulation trials. CyHMMs have lower mean error than all baselines. Relative error improvement of CyHMM over baselines shown in parentheses. We order methods by their mean error. Partial Periodicity Detection can only be applied to binary data.</span>
          </div>
          <table class="table">
            <thead>
              <tr>
                <th style="text-align:left;"></th>
                <th style="text-align:left;">All trials</th>
                <th style="text-align:left;">Binary</th>
                <th style="text-align:left;">Continuous</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td style="text-align:left;"><strong>CyHMM</strong></td>
                <td style="text-align:left;"><strong>2.5 days</strong></td>
                <td style="text-align:left;"><strong>3.1 days</strong></td>
                <td style="text-align:left;"><strong>1.9 days</strong></td>
              </tr>
              <tr>
                <td style="text-align:left;">Autocorrelation</td>
                <td style="text-align:left;">5.9 (58%)</td>
                <td style="text-align:left;">6.8 (55%)</td>
                <td style="text-align:left;">4.9 (62%)</td>
              </tr>
              <tr>
                <td style="text-align:left;">ARSER</td>
                <td style="text-align:left;">10.5 (76%)</td>
                <td style="text-align:left;">13.7 (78%)</td>
                <td style="text-align:left;">7.2 (74%)</td>
              </tr>
              <tr>
                <td style="text-align:left;">Fourier</td>
                <td style="text-align:left;">13.3 (81%)</td>
                <td style="text-align:left;">16.0 (81%)</td>
                <td style="text-align:left;">10.6 (83%)</td>
              </tr>
              <tr>
                <td style="text-align:left;">Partial Periodicity Detection</td>
                <td style="text-align:left;">Binary only</td>
                <td style="text-align:left;">19.4 (84%)</td>
                <td style="text-align:left;">Binary only</td>
              </tr>
            </tbody>
          </table>
        </div>
        <p>When we examine the effects of altering specific simulation parameters, we find that CyHMM performance, as measured by correlation with true cycle length, improves as noise decreases, fraction of missing data decreases, maximum time series length increases, variability in cycle length <em>between</em> individuals increases, and variability in cycle length <em>within</em> individuals decreases. We also find that the superior performance of CyHMMs over baselines occurs in part because CyHMMs are more robust to noise and missing data. In very low noise settings, several baselines perform comparably to CyHMMs (autocorrelation, ARSER, and Fourier), but baseline performance rapidly degrades in more realistic scenarios as noise increases. It is logical that CyHMMs would be more robust to noise, because they share information across individuals and across features, while baselines do not. Similarly, CyHMMs should be more robust to missing data because they explicitly incorporate it into the generative model and use it to infer cycle state. We also note that it is striking that CyHMMs can outperform ARSER and Fourier transforms even though both are based on sinusoidal models, and our simulated data is drawn from sinusoids.</p>
      </section>
      <section id="sec-14">
        <header>
          <div class="title-info">
            <h3><span class="section-number">5.3</span> Inference of feature trajectories</h3>
          </div>
        </header>
        <p>We compute feature trajectories and variabilities as described in Section <a class="sec" href="#sec-10">4.3</a>, and assess how well the variabilities computed from the inferred trajectories correlate with the variabilities computed from the true trajectories. (To compute the true trajectories, we compute the mean value of each symptom on each cycle day.) The mean correlation between the true and inferred variabilities is 0.97 across all binary simulations and 0.98 across all continuous simulations, demonstrating that CyHMMs are able to correctly infer feature variabilities. Because the baselines discussed in Section <a class="sec" href="#sec-13">5.2</a> are not generative models designed to infer feature trajectories, we do not compare to them.</p>
      </section>
    </section>
    <section id="sec-15">
      <header>
        <div class="title-info">
          <h2><span class="section-number">6</span> Evaluation on Real-World Data</h2>
        </div>
      </header>
      <p>We demonstrate the applicability of CyHMMs by using them to infer cycle characteristics in two real-world health datasets: a menstrual cycles dataset, and an exercise and activity tracking dataset. We choose these datasets for two reasons: first, they have information on true cycle state, allowing us to compare our model to existing baselines, and second, they track important health-related features<a class="fn" href="#fn6" id="foot-fn6"><sup>6</sup></a>.</p>
      <section id="sec-16">
        <header>
          <div class="title-info">
            <h3><span class="section-number">6.1</span> Datasets</h3>
          </div>
        </header>
        <p><strong>Menstrual cycles dataset.</strong> The human menstrual cycle is fundamental to women's health; features of the cycle have been linked to cancer [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0023">23</a>], depression [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0019">19</a>], and sports injuries [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0059">59</a>], and the cycle has been proposed as a vital sign [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0004">4</a>]. We obtained data from one of the most popular menstrual cycle tracking apps. Upon logging into the app, users can record <em>binary</em> features including period/cycle start (<em>e.g.</em>, light, moderate, or heavy bleeding), mood (<em>e.g.</em>, happy, sad, or sensitive), pain (<em>e.g.</em>, cramps or headache) or sexual activity (<em>e.g.</em>, protected or unprotected sex). We define the start of each menstrual cycle as the start of period bleeding. We confirmed that statistics of the dataset were consistent with existing literature on the menstrual cycle: the average cycle duration was 28 days, consistent with previous investigations, and features that peaked before the cycle start were consistent with previous investigations of premenstrual syndrome [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0050">50</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0064">64</a>].</p>
        <p>This dataset has a number of traits which make it an useful test case for cycle inference methods. It has a rich set of features for each user, a large number of users, and variation in cycles both between and within users: menstrual cycle length varies from person to person, and even from cycle to cycle, as do cycle symptoms. Most importantly, it contains ground truth on the true cycle start times for each user (as indicated by bleeding). While CyHMMs are more generally applicable to cases when the true cycle starts are not known, true cycle starts are essential for validating that the method works and comparing it to other methods.</p>
        <p>We fit a CyHMM using features shown by the tracking app's default five categories—mood, sleep, sex, pain, and energy—because many of the other features had very sparse data. Importantly, we did not provide the model with any features directly related to cycle start, like bleeding, so the model was given no information about when cycles started. In total this left us with 19 binary features for each day. We filtered for users that logged a feature using the app at least 20 times and had a timespan of at least 50 days where they used the app regularly (logging at least once every two days). After filtering, our analysis included 22 million observations (<em>i.e.</em>, measurements of one feature for one user at one timestep) from 9,885 app users. We fit a CyHMM with four hidden states because the menstrual cycle is often divided into four latent hormonal phases [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0007">7</a>]. (To study the robustness of our model to this parameter, we also examined results using three or five states.) If a user logged no features on a given day, we considered data for that day to be unobserved because, as explained above, we did not know whether the user truly had no symptoms or merely neglected to log them.</p>
        <p><strong>Activity tracking dataset.</strong> We obtained data from a large activity tracking mobile app that has been previously used in studies of human health&nbsp;[<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0002">2</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0003">3</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0055">55</a>]. For each user on each day, our dataset included sleep start time, sleep end time, steps taken, and total calories burned. Each feature had missing data caused, for example, by a user forgetting to log their sleep. Our analysis included all regular app users that logged into the app to record a measurement at least 20 times, had a timespan of at least 50 days where they used the app regularly (logging at least once every two days) and had less than 50% missing data in every feature; in total, after filtering, our data consisted of 8 million observations from 6,882 users. Because we are interested in cycles, we removed long-term time trends (<em>e.g.</em>, people using the app often lose weight) by subtracting off the centered moving average using a two-week window around each day. This is analogous to the trend-cycle decomposition often used in time-series analysis [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0020">20</a>].</p>
        <p>Previous research has found that people's activity patterns show weekly cycles: for example, they sleep and wake later on the weekends [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0043">43</a>]. We thus assessed how well CyHMMs could model this weekend effect from time series data without being provided with the day of the week: essentially, testing whether CyHMMs could recover weekly cycle patterns without knowing the day of the week. We fit a Gaussian CyHMM with two states to correspond to weekday and weekend using sleep start time, sleep end time, steps taken, and calories burned as continuous features. We verified that these features did in fact show weekly cycles in the data: individuals go to bed and wake up later on average on the weekends, with steps taken and calories burned remaining fairly constant across weekends and weekdays.</p>
      </section>
      <section id="sec-17">
        <header>
          <div class="title-info">
            <h3><span class="section-number">6.2</span> Analysis of menstrual cycles</h3>
          </div>
        </header>
        <p>We used ground truth data to evaluate how well CyHMMs could recover basic features of the menstrual cycle: cycle length, trajectory of each feature over the course of the cycle, most variable features, and clusters of individuals with different symptom patterns.</p>
        <div class="table-responsive" id="tab2">
          <div class="table-caption">
            <span class="table-number">Table 2:</span> <span class="table-title">Comparison of our model to baselines on menstrual cycle data for users who recorded five or more cycles. Errors reported are averaged across all users; the error for each user is the difference between the true cycle length for each user and the inferred cycle length for that user. We order methods by their mean error.</span>
          </div>
          <table class="table">
            <thead>
              <tr>
                <th style="text-align:left;">Method</th>
                <th style="text-align:left;">Population cycle length</th>
                <th style="text-align:left;">Median error</th>
                <th style="text-align:left;">Mean error</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td style="text-align:left;">Ground truth</td>
                <td style="text-align:left;">28.0 days</td>
                <td style="text-align:left;">0.0 days</td>
                <td style="text-align:left;">0.0 days</td>
              </tr>
              <tr>
                <td style="text-align:left;"><strong>CyHMM</strong></td>
                <td style="text-align:left;"><strong>29.0</strong></td>
                <td style="text-align:left;"><strong>2.0</strong></td>
                <td style="text-align:left;"><strong>3.6</strong></td>
              </tr>
              <tr>
                <td style="text-align:left;">Autocorrelation</td>
                <td style="text-align:left;">19.0</td>
                <td style="text-align:left;">9.0</td>
                <td style="text-align:left;">9.9</td>
              </tr>
              <tr>
                <td style="text-align:left;">Partial Periodicity Detection</td>
                <td style="text-align:left;">21.9</td>
                <td style="text-align:left;">8.4</td>
                <td style="text-align:left;">10.0</td>
              </tr>
              <tr>
                <td style="text-align:left;">Fourier transform</td>
                <td style="text-align:left;">14.8</td>
                <td style="text-align:left;">12.7</td>
                <td style="text-align:left;">11.9</td>
              </tr>
              <tr>
                <td style="text-align:left;">ARSER</td>
                <td style="text-align:left;">20.9</td>
                <td style="text-align:left;">13.3</td>
                <td style="text-align:left;">12.7</td>
              </tr>
            </tbody>
          </table>
        </div>
        <section id="sec-18">
          <p><em>6.2.1 Inference of cycle length.</em> As with simulated data, we assessed how well CyHMMs could infer the true cycle length for each user on the menstrual cycles dataset, defining the true cycle length for each user to be their average gap between period starts. To ensure we had reliable data for each user, we assessed performance on users who recorded at least 5 cycles. We compared our model to the baselines described in Section <a class="sec" href="#sec-13">5.2</a>. CyHMMs inferred the cycle length for each user more accurately than did the baselines (Table&nbsp;<a class="tbl" href="#tab2">2</a>) and were also closer to the population average cycle length<a class="fn" href="#fn7" id="foot-fn7"><sup>7</sup></a>. CyHMMs had a mean error that was 63% lower, and a median error that was 78% lower, than autocorrelation, the baseline with the lowest mean error.</p>
          <p><strong>Inference of feature trajectories.</strong> We inferred feature trajectories using the methodology described in Section &nbsp;<a class="sec" href="#sec-10">4.3</a>. In Figure &nbsp;<a class="fig" href="#fig3">3</a> we visualize the inferred feature trajectories for pain, emotion, and sleep features. To provide context for the trajectories, we also plot the true cycle start day (when period bleeding starts), although this information is not given to the model. The model correctly infers that mood and pain features increase just before the cycle starts, consistent with existing literature on premenstrual syndrome [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0050">50</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0064">64</a>]. Finally, the model infers that users are more likely to log in to record features when their cycle is starting; this is logical because many people use the app to track cycle starts, and speaks to the importance of modeling missing data.</p>
          <figure id="fig3">
            <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186052/images/www2018-61-fig3.jpg" class="img-responsive" alt="Figure 3" longdesc="" />
            <div class="figure-caption">
              <span class="figure-number">Figure 3:</span> <span class="figure-title">Model-inferred feature trajectories for a subset of menstrual cycle features. The vertical axis is the relative probability of emitting a feature (conditional on non-missing data); the horizontal axis is cycle day. For context, the vertical black line shows the day of the true cycle start, although this information is not given to the model. We center cycle start for ease of visualization. Pain and negative mood features increase just prior to cycle start; sleep features do not show consistent patterns. The relative probability that <em>no</em> features are logged (“No features logged”) decreases near cycle start, because many people use the app to track their cycle starts.</span>
            </div>
          </figure>
          <p></p>
          <p><strong>Feature variability.</strong> We also found that CyHMMs correctly recovered which features showed the most variability over the course of the cycle. We compare the variabilities computed from the true feature trajectories to the variabilities computed from the inferred feature trajectories as described in Section <a class="sec" href="#sec-10">4.3</a>. (We computed the true feature trajectories by aligning all users to their period starts, and computing the fraction of users experiencing a given feature on each day relative to period start). The true variabilities were highly correlated with the inferred variabilities (<em>r</em> = 0.86): <em>e.g.</em>, the model correctly recovered the fact that pain features showed greater variability than sleep features.</p>
          <p><strong>Clustering of similar individuals.</strong> We divided individuals into five clusters with similar feature patterns using the methodology described in Section <a class="sec" href="#sec-10">4.3</a>. We chose the number of clusters using the elbow method [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0057">57</a>]: beyond five clusters, both the train and test log likelihoods increased much more slowly. Our clustering correlates with individual-specific features not given to the model (Table&nbsp;<a class="tbl" href="#tab3">3</a>; all features below the double horizontal line), like medical history and age. We report only features which show statistically significant differences (categorical F-test <em>p</em> &lt; 0.001), revealing distinct groups:</p>
          <ul class="list-no-style">
            <li id="list16" label="•">
              <strong>Cluster 1: Severe symptom users.</strong> This cluster (Table&nbsp;<a class="tbl" href="#tab3">3</a>, first column), is most likely to report going to the doctor or gynecologist or taking pain medication (rows below double horizontal line). Our model is able to identify this subcluster even though it is given no information on doctors’ visits. These users also have the highest rate of pain features, are more likely to report negative emotions and less sleep, and are most likely to report having a high sex drive (rows above double horizontal line).<br />
            </li>
            <li id="list17" label="•"><strong>Cluster 2: Birth control users.</strong> These users primarily use the app to keep track of whether they have taken birth control (“Logged birth control pill” row)—again, not information given to the model. Hence, they are less likely to log other features and have a high fraction of missing data (“Fraction of days with no data” row), but likely to be on birth control. Consistent with this, they have a higher rate of having unprotected sex than do other users, and the lowest rate of positive pregnancy tests.<br /></li>
            <li id="list18" label="•"><strong>Cluster 3: Happy users.</strong> This group of users is most likely to report positive features like happy emotion (top row), high energy, and sleeping 6-9 hours.<br /></li>
            <li id="list19" label="•"><strong>Cluster 4: Low-emotion users.</strong> This group of users is relatively unlikely to log any emotion features (top three rows).<br /></li>
            <li id="list20" label="•"><strong>Cluster 5: Infrequent loggers.</strong> This cluster of users appears to put less effort into using the app; when they record features, they record the fewest of any group besides the birth control group (“Features logged per non-missing day” row).<br /></li>
          </ul>
          <div class="table-responsive" id="tab3">
            <div class="table-caption">
              <span class="table-number">Table 3:</span> <span class="table-title">Clustering of menstrual cycle data. Only the top set of user characteristics (above the double line) is accessible by the model, but the clustering also correlates with differences in medical history, age, and true cycle length (bottom set of characteristics) which are not provided to the model. All differences between clusters are statistically significant (<em>p</em> &lt; 0.001, categorical F-test). The largest value in each row is shown in bold.</span>
            </div>
            <table class="table">
              <thead>
                <tr>
                  <th style="text-align:left;">Cluster</th>
                  <th style="text-align:left;">1</th>
                  <th style="text-align:left;">2</th>
                  <th style="text-align:left;">3</th>
                  <th style="text-align:left;">4</th>
                  <th style="text-align:left;">5</th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td style="text-align:left;">Emotion:happy (% of time)</td>
                  <td style="text-align:left;">55%</td>
                  <td style="text-align:left;">15%</td>
                  <td style="text-align:left;"><strong>74%</strong></td>
                  <td style="text-align:left;">24%</td>
                  <td style="text-align:left;">66%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Emotion:sad</td>
                  <td style="text-align:left;"><strong>22%</strong></td>
                  <td style="text-align:left;">9%</td>
                  <td style="text-align:left;">11%</td>
                  <td style="text-align:left;">9%</td>
                  <td style="text-align:left;">17%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Emotion:sensitive</td>
                  <td style="text-align:left;"><strong>46%</strong></td>
                  <td style="text-align:left;">15%</td>
                  <td style="text-align:left;">27%</td>
                  <td style="text-align:left;">20%</td>
                  <td style="text-align:left;">32%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">High energy</td>
                  <td style="text-align:left;">36%</td>
                  <td style="text-align:left;">7%</td>
                  <td style="text-align:left;"><strong>53%</strong></td>
                  <td style="text-align:left;">16%</td>
                  <td style="text-align:left;">4%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Low energy</td>
                  <td style="text-align:left;"><strong>43%</strong></td>
                  <td style="text-align:left;">11%</td>
                  <td style="text-align:left;">33%</td>
                  <td style="text-align:left;">23%</td>
                  <td style="text-align:left;">4%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Pain:cramps</td>
                  <td style="text-align:left;"><strong>20%</strong></td>
                  <td style="text-align:left;"><strong>20%</strong></td>
                  <td style="text-align:left;">12%</td>
                  <td style="text-align:left;">12%</td>
                  <td style="text-align:left;">15%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Pain:headache</td>
                  <td style="text-align:left;"><strong>33%</strong></td>
                  <td style="text-align:left;">14%</td>
                  <td style="text-align:left;">15%</td>
                  <td style="text-align:left;">14%</td>
                  <td style="text-align:left;">17%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Pain:tender breasts</td>
                  <td style="text-align:left;"><strong>18%</strong></td>
                  <td style="text-align:left;">11%</td>
                  <td style="text-align:left;">9%</td>
                  <td style="text-align:left;">9%</td>
                  <td style="text-align:left;">12%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">High sex drive</td>
                  <td style="text-align:left;"><strong>20%</strong></td>
                  <td style="text-align:left;">7%</td>
                  <td style="text-align:left;">6%</td>
                  <td style="text-align:left;">6%</td>
                  <td style="text-align:left;">11%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Had protected sex</td>
                  <td style="text-align:left;">5%</td>
                  <td style="text-align:left;"><strong>15%</strong></td>
                  <td style="text-align:left;">4%</td>
                  <td style="text-align:left;">5%</td>
                  <td style="text-align:left;">5%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Had unprotected sex</td>
                  <td style="text-align:left;">10%</td>
                  <td style="text-align:left;"><strong>25%</strong></td>
                  <td style="text-align:left;">4%</td>
                  <td style="text-align:left;">8%</td>
                  <td style="text-align:left;">6%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Had withdrawal sex</td>
                  <td style="text-align:left;"><strong>16%</strong></td>
                  <td style="text-align:left;">8%</td>
                  <td style="text-align:left;">2%</td>
                  <td style="text-align:left;">3%</td>
                  <td style="text-align:left;">7%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Slept 0-3 hours</td>
                  <td style="text-align:left;"><strong>5%</strong></td>
                  <td style="text-align:left;">1%</td>
                  <td style="text-align:left;">1%</td>
                  <td style="text-align:left;">1%</td>
                  <td style="text-align:left;">0%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Slept 3-6 hours</td>
                  <td style="text-align:left;"><strong>40%</strong></td>
                  <td style="text-align:left;">8%</td>
                  <td style="text-align:left;">18%</td>
                  <td style="text-align:left;">21%</td>
                  <td style="text-align:left;">2%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Slept 6-9 hours</td>
                  <td style="text-align:left;">39%</td>
                  <td style="text-align:left;">13%</td>
                  <td style="text-align:left;"><strong>63%</strong></td>
                  <td style="text-align:left;">52%</td>
                  <td style="text-align:left;">4%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Slept 9+ hours</td>
                  <td style="text-align:left;">10%</td>
                  <td style="text-align:left;">3%</td>
                  <td style="text-align:left;"><strong>13%</strong></td>
                  <td style="text-align:left;">9%</td>
                  <td style="text-align:left;">1%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Features logged per non-missing day</td>
                  <td style="text-align:left;"><strong>4.6</strong></td>
                  <td style="text-align:left;">1.9</td>
                  <td style="text-align:left;">3.7</td>
                  <td style="text-align:left;">2.5</td>
                  <td style="text-align:left;">2.1</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Fraction of days with no data</td>
                  <td style="text-align:left;">5%</td>
                  <td style="text-align:left;"><strong>68%</strong></td>
                  <td style="text-align:left;">4%</td>
                  <td style="text-align:left;">18%</td>
                  <td style="text-align:left;">10%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Recorded doctor's appt</td>
                  <td style="text-align:left;"><strong>37%</strong></td>
                  <td style="text-align:left;">15%</td>
                  <td style="text-align:left;">29%</td>
                  <td style="text-align:left;">27%</td>
                  <td style="text-align:left;">15%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Recorded ob/gyn appt</td>
                  <td style="text-align:left;"><strong>24%</strong></td>
                  <td style="text-align:left;">14%</td>
                  <td style="text-align:left;">17%</td>
                  <td style="text-align:left;">18%</td>
                  <td style="text-align:left;">11%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Took pain medication</td>
                  <td style="text-align:left;"><strong>36%</strong></td>
                  <td style="text-align:left;">17%</td>
                  <td style="text-align:left;">27%</td>
                  <td style="text-align:left;">27%</td>
                  <td style="text-align:left;">16%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Logged birth control pill</td>
                  <td style="text-align:left;">38%</td>
                  <td style="text-align:left;"><strong>85%</strong></td>
                  <td style="text-align:left;">29%</td>
                  <td style="text-align:left;">49%</td>
                  <td style="text-align:left;">33%</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Negative pregnancy test</td>
                  <td style="text-align:left;">15%</td>
                  <td style="text-align:left;">13%</td>
                  <td style="text-align:left;">9%</td>
                  <td style="text-align:left;">13%</td>
                  <td style="text-align:left;"><strong>16%</strong></td>
                </tr>
                <tr>
                  <td style="text-align:left;">Positive pregnancy test</td>
                  <td style="text-align:left;">9%</td>
                  <td style="text-align:left;">5%</td>
                  <td style="text-align:left;">6%</td>
                  <td style="text-align:left;">6%</td>
                  <td style="text-align:left;"><strong>10%</strong></td>
                </tr>
                <tr>
                  <td style="text-align:left;">Age</td>
                  <td style="text-align:left;">21.9</td>
                  <td style="text-align:left;">21.7</td>
                  <td style="text-align:left;">20.6</td>
                  <td style="text-align:left;"><strong>23.9</strong></td>
                  <td style="text-align:left;">20.8</td>
                </tr>
                <tr>
                  <td style="text-align:left;">Cycle length, days</td>
                  <td style="text-align:left;">28.2</td>
                  <td style="text-align:left;">29.0</td>
                  <td style="text-align:left;">28.8</td>
                  <td style="text-align:left;">28.3</td>
                  <td style="text-align:left;"><strong>29.4</strong></td>
                </tr>
              </tbody>
            </table>
          </div>
          <p>The fact that the clustering correlates with features not used in the clustering—like birth control, doctors’ appointments, and age—indicates that CyHMMs are correctly identifying true population heterogeneity. For example, the model is able to identify “users more likely to be on birth control” (information it is not given) as “users with a high probability of missing data” (something it explicitly models). Recovering the birth control subpopulation is essential for accurately modeling the menstrual cycle. Previous work has found that women on birth control experience different menstrual cycle symptom progression—for example, less variability in mood over the course of the cycle [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0045">45</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0048">48</a>]. (We note that the fact that women in the birth control cluster are relatively unlikely to bother logging emotion symptoms is consistent with this lack of variability.) Recovering the birth control subpopulation is also important for designing health interventions. For example, women on birth control require different pharmacologic treatment [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0030">30</a>] and, because their hormone cycles are different, have different risks of hormone-related cancers [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0044">44</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0046">46</a>].</p>
        </section>
      </section>
      <section id="sec-19">
        <header>
          <div class="title-info">
            <h3><span class="section-number">6.3</span> Analysis of cycles in activity tracking</h3>
          </div>
        </header>
        <p><strong>Inference of population cycle length.</strong>We define the model-inferred cycle lengths as in Section <a class="sec" href="#sec-10">4.3</a>. Since there are no ground truth cycle lengths for all individuals in this dataset—unlike with the menstrual cycles dataset, there is no clear physiological feature with which to define cycle starts—a supervised comparison to baselines is impossible in this setting. Instead, we directly evaluate the plausibility of CyHMM cycle length inference. When we apply CyHMMs to our dataset, the model infers that the most frequent cycle length in the population as a whole is 7 days (Figure&nbsp;<a class="fig" href="#fig4">4</a>). Inspecting the model, the two learned hidden states also clearly correspond to weekday and weekend, even though day of the week is not provided to the model: the “weekend hidden state” has an inferred average duration of 2.2 days (as compared to the true weekend length of 2 days), while the “weekday hidden state” has an inferred average duration of 4.9 days (as compared to the true weekday length of 5 days). The weekend hidden state has a later inferred mean for sleep start and sleep end than the weekday hidden state, consistent with the fact that individuals sleep later on the weekends. On timesteps when the model infers an individual is in the weekend state, it is twice as likely to be the true weekend. All these facts indicate that the model is successfully recovering weekday/weekend cycles. This is consistent with previous findings that human activity shows weekly cycles [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0043">43</a>].</p>
        <figure id="fig4">
          <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186052/images/www2018-61-fig4.jpg" class="img-responsive" alt="Figure 4" longdesc="" />
          <div class="figure-caption">
            <span class="figure-number">Figure 4:</span> <span class="figure-title">Histogram of cycle lengths for individuals in the activity tracking dataset. The cycle length for each individual is the modal cycle length for that individual.</span>
          </div>
        </figure><strong>Clustering of similar individuals.</strong>We next investigated whether CyHMMs could cluster individuals into groups that correlated with fundamental features like BMI, age, and gender without being provided with these features to use in the clustering. Our clustering shows statistically significant correlations with all these features (<em>p</em> &lt; 0.001, categorical F-test). For example, we identify one cluster with the highest proportion of males (78%) with the highest BMIs (26.9); another cluster, essentially its opposite, has the lowest proportion of males (48%) and the lowest BMIs (25.4). (Cluster differences between BMIs remain significant when adjusting for age and sex via linear regression.) None of these features are given to the CyHMM in performing the clustering, indicating that the model is successfully identifying true population heterogeneity. Our finding is consistent with previous work that finds that weekend-weekday sleep differences correlate with important health metrics like body mass index (BMI) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0053">53</a>]. CyHMMs can thus identify subpopulations that differ along fundamental health metrics like BMI, potentially allowing for unsupervised discovery of behavioral risk factors to inform monitoring of obesity.
        <p></p>
      </section>
    </section>
    <section id="sec-20">
      <header>
        <div class="title-info">
          <h2><span class="section-number">7</span> Conclusion</h2>
        </div>
      </header>
      <p>Modeling cycles in time series data is important for understanding human health, but it is challenging because cycle starts are rarely labeled. Here we present CyHMMs, which take as input a multivariate time series for each individual in a population, infer what latent cycle state an individual is in at each timestep, and use this inferred state to recover fundamental cycle characteristics. Evaluating our method on both simulated data and two real-world datasets with ground-truth information on cycle states, we find CyHMMs infer the true cycle length more accurately than do baselines. CyHMMs can also infer cycle characteristics that baselines cannot: we show they can infer how each observed feature changes over the course of the cycle, accurately recover the most variable features, and find clusters of individuals with distinct cycle patterns. While we evaluated CyHMMs on datasets in which cycle starts are known in order to validate our method, they are designed to be applied to datasets on which cycle starts are unknown: for example, sentiment in individual Twitter feeds.</p>
      <p>CyHMMs assume that observed data is generated by a single latent state that progresses cyclically. Future work could attempt to relax this assumption while retaining the interpretability it provides. Natural extensions to the CyHMM model might retain its core idea of a cyclic latent state while increasing the model's expressive power: for example, by using a multidimensional hidden state as factorial HMMs [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0024">24</a>] or LSTMs [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0031">31</a>] do, with some cyclic and some unconstrained dimensions. A multidimensional hidden state could capture data that includes multiple cycles, or both a cycle and a time trend. These extensions flow from the fact that CyHMMs, by using a generative model with hidden state, provide a natural and flexible way to model cycles fundamental to human health.</p>
      <p><strong>Code availability.</strong> <a href="https://github.com/epierson9/cyclic_HMMs" target="_blank">https://github.com/epierson9/cyclic_HMMs</a>.</p>
      <p><strong>Acknowledgments.</strong> We thank David Hallac, Chris Olah, Nat Roth, Marinka Zitnik, the apps that provided data, and the reviewers for their valuable feedback. E.P. was supported by Hertz and NDSEG Fellowships. T.A. was supported by National Institutes of Health (NIH) grant U54 EB020405 and a SAP Stanford Graduate Fellowship.</p>
    </section>
  </section>
  <section class="back-matter">
    <section id="ref-001">
      <header>
        <div class="title-info">
          <h2 class="page-brake-head">REFERENCES</h2>
        </div>
      </header>
      <ul class="bibUl">
        <li id="BibPLXBIB0001" label="[1]">T.&nbsp;Althoff, E.&nbsp;Horvitz, R.&nbsp;W. White, and J.&nbsp;Zeitzer. Harnessing the web for population-scale physiological sensing: A case study of sleep and performance. <em>WWW</em>, 2017.</li>
        <li id="BibPLXBIB0002" label="[2]">T.&nbsp;Althoff, P.&nbsp;Jindal, and J.&nbsp;Leskovec. Online actions with offline impact: How online social networks influence online and offline user behavior. In <em>WSDM</em>, 2017.</li>
        <li id="BibPLXBIB0003" label="[3]">T.&nbsp;Althoff, R.&nbsp;Sosic, J.&nbsp;L. Hicks, A.&nbsp;C. King, S.&nbsp;L. Delp, and J.&nbsp;Leskovec. Large-scale physical activity data reveal worldwide activity inequality. <em>Nature</em>, 2017.</li>
        <li id="BibPLXBIB0004" label="[4]">American Academy of Pediatrics, American College of Obstetricians and Gynecologists, et&nbsp;al. Menstruation in girls and adolescents: using the menstrual cycle as a vital sign. <em>Pediatrics</em>, 2006.</li>
        <li id="BibPLXBIB0005" label="[5]">W.&nbsp;G. Aref, M.&nbsp;G. Elfeky, and A.&nbsp;K. Elmagarmid. Incremental, online, and merge mining of partial periodic patterns in time-series databases. <em>TKDE</em>, 2004.</li>
        <li id="BibPLXBIB0006" label="[6]">C.&nbsp;Berberidis, I.&nbsp;Vlahavas, W.&nbsp;G. Aref, M.&nbsp;Atallah, and A.&nbsp;K. Elmagarmid. On the discovery of weak periodicities in large time series. In <em>PKDD</em>, 2002.</li>
        <li id="BibPLXBIB0007" label="[7]">S.&nbsp;L. Berga and S.&nbsp;Yen. Circadian pattern of plasma melatonin concentrations during four phases of the human menstrual cycle. <em>Neuroendocrinology</em>, 1990.</li>
        <li id="BibPLXBIB0008" label="[8]">J.&nbsp;A. Bilmes et&nbsp;al. A gentle tutorial of the EM algorithm and its application to parameter estimation for gaussian mixture and hidden Markov models. <em>International Computer Science Institute</em>, 1998.</li>
        <li id="BibPLXBIB0009" label="[9]">R.&nbsp;N. Bracewell and R.&nbsp;N. Bracewell. <em>The Fourier transform and its applications</em>. 1986.</li>
        <li id="BibPLXBIB0010" label="[10]">C.&nbsp;E. Brown. Coefficient of variation. In <em>Applied multivariate statistics in geohydrology and related sciences</em>. 1998.</li>
        <li id="BibPLXBIB0011" label="[11]">G.&nbsp;Celeux and J.-B. Durand. Selecting hidden Markov model state number with cross-validated likelihood. <em>Computational Statistics</em>, 2008.</li>
        <li id="BibPLXBIB0012" label="[12]">A.&nbsp;K. Chanda, S.&nbsp;Saha, M.&nbsp;A. Nishi, M.&nbsp;Samiullah, and C.&nbsp;F. Ahmed. An efficient approach to mine flexible periodic patterns in time series databases. <em>Engineering Applications of Artificial Intelligence</em>, 2015.</li>
        <li id="BibPLXBIB0013" label="[13]">P.&nbsp;Chaovalit, A.&nbsp;Gangopadhyay, G.&nbsp;Karabatis, and Z.&nbsp;Chen. Discrete wavelet transform-based time series analysis and mining. <em>ACM Computing Surveys (CSUR)</em>, 2011.</li>
        <li id="BibPLXBIB0014" label="[14]">L.&nbsp;Chiazze, F.&nbsp;T. Brayer, J.&nbsp;J. Macisco, M.&nbsp;P. Parker, and B.&nbsp;J. Duffy. The length and variability of the human menstrual cycle. <em>JAMA</em>, 1968.</li>
        <li id="BibPLXBIB0015" label="[15]">I.&nbsp;Daubechies. The wavelet transform, time-frequency localization and signal analysis. <em>Transactions on Information Theory</em>, 1990.</li>
        <li id="BibPLXBIB0016" label="[16]">A.&nbsp;Deckard, R.&nbsp;C. Anafi, J.&nbsp;B. Hogenesch, S.&nbsp;B. Haase, and J.&nbsp;Harer. Design and analysis of large-scale biological rhythm studies: a comparison of algorithms for detecting periodic signals in biological data. <em>Bioinformatics</em>, 2013.</li>
        <li id="BibPLXBIB0017" label="[17]">M.&nbsp;Dewar, C.&nbsp;Wiggins, and F.&nbsp;Wood. Inference in hidden Markov models with explicit state duration distributions. <em>IEEE Signal Process. Lett.</em>, 2012.</li>
        <li id="BibPLXBIB0018" label="[18]">D.&nbsp;F. Dinges. An overview of sleepiness and accidents. <em>Journal of Sleep Research</em>, 1995.</li>
        <li id="BibPLXBIB0019" label="[19]">J.&nbsp;Endicott. The menstrual cycle and mood disorders. <em>Journal of Affective Disorders</em>, 1993.</li>
        <li id="BibPLXBIB0020" label="[20]">W.&nbsp;Fellner. Trends and cycles in economic activity. 1956.</li>
        <li id="BibPLXBIB0021" label="[21]">J.&nbsp;Ferguson. Variable duration models for speech. In <em>Proc. Symp. on the Application of Hidden Markov Models to Text and Speech</em>, 1980.</li>
        <li id="BibPLXBIB0022" label="[22]">R.&nbsp;L. Findling, B.&nbsp;L. Gracious, N.&nbsp;K. McNamara, E.&nbsp;A. Youngstrom, C.&nbsp;A. Demeter, L.&nbsp;A. Branicky, and J.&nbsp;R. Calabrese. Rapid, continuous cycling and psychiatric co-morbidity in pediatric bipolar I disorder. <em>Bipolar disorders</em>, 2001.</li>
        <li id="BibPLXBIB0023" label="[23]">M.&nbsp;Garland et&nbsp;al. Menstrual cycle characteristics and history of ovulatory infertility in relation to breast cancer risk in a large cohort of US women. <em>American Journal of Epidemiology</em>, 1998.</li>
        <li id="BibPLXBIB0024" label="[24]">Z.&nbsp;Ghahramani and M.&nbsp;I. Jordan. Factorial hidden Markov models. In <em>NIPS</em>, 1996.</li>
        <li id="BibPLXBIB0025" label="[25]">C.&nbsp;Giannella, J.&nbsp;Han, J.&nbsp;Pei, X.&nbsp;Yan, and P.&nbsp;S. Yu. Mining frequent patterns in data streams at multiple time granularities. <em>Next Generation Data Mining</em>, 2003.</li>
        <li id="BibPLXBIB0026" label="[26]">S.&nbsp;A. Golder and M.&nbsp;W. Macy. Diurnal and seasonal mood vary with work, sleep, and daylength across diverse cultures. <em>Science</em>, 2011.</li>
        <li id="BibPLXBIB0027" label="[27]">J.&nbsp;Han, G.&nbsp;Dong, and Y.&nbsp;Yin. Efficient mining of partial periodic patterns in time series database. <em>ICDE</em>, 1999.</li>
        <li id="BibPLXBIB0028" label="[28]">J.&nbsp;Han, W.&nbsp;Gong, and Y.&nbsp;Yin. Mining segment-wise periodic patterns in time-related databases. In <em>KDD</em>, 1998.</li>
        <li id="BibPLXBIB0029" label="[29]">J.&nbsp;Han, J.&nbsp;Pei, Y.&nbsp;Yin, and R.&nbsp;Mao. Mining frequent patterns without candidate generation: A frequent-pattern tree approach. <em>DMKD</em>, 2004.</li>
        <li id="BibPLXBIB0030" label="[30]">T.&nbsp;Hassan. Pharmacologic considerations for patients taking oral contraceptives. <em>Connecticut Dental Student Journal</em>, 1987.</li>
        <li id="BibPLXBIB0031" label="[31]">S.&nbsp;Hochreiter and J.&nbsp;Schmidhuber. Long short-term memory. <em>Neural Computation</em>, 1997.</li>
        <li id="BibPLXBIB0032" label="[32]">Y.-H. Hu, C.-F. Tsai, C.-T. Tai, and I.-C. Chiang. A novel approach for mining cyclically repeated patterns with multiple minimum supports. <em>Applied Soft Computing</em>, 2015.</li>
        <li id="BibPLXBIB0033" label="[33]">M.&nbsp;E. Hughes, J.&nbsp;B. Hogenesch, and K.&nbsp;Kornacker. JTK_CYCLE: an efficient nonparametric algorithm for detecting rhythmic components in genome-scale data sets. <em>Journal of Biological Rhythms</em>, 2010.</li>
        <li id="BibPLXBIB0034" label="[34]">R.&nbsp;U. Kiran, H.&nbsp;Shang, M.&nbsp;Toyoda, and M.&nbsp;Kitsuregawa. Discovering recurring patterns in time series. In <em>EDBT</em>, 2015.</li>
        <li id="BibPLXBIB0035" label="[35]">O.&nbsp;Kostakis, N.&nbsp;Tatti, and A.&nbsp;Gionis. Discovering recurring activity in temporal networks. <em>DMKD</em>, 2017.</li>
        <li id="BibPLXBIB0036" label="[36]">E.&nbsp;A. Lenton, B.&nbsp;Landgren, and L.&nbsp;Sexton. Normal variation in the length of the luteal phase of the menstrual cycle: identification of the short luteal phase. <em>British Journal of Obstetrics and Gynecology</em>, 1984.</li>
        <li id="BibPLXBIB0037" label="[37]">S.&nbsp;E. Levinson. Continuously variable duration hidden Markov models for automatic speech recognition. <em>Computer Speech &amp; Language</em>, 1986.</li>
        <li id="BibPLXBIB0038" label="[38]">L.&nbsp;Li and B.&nbsp;A. Prakash. Time series clustering: Complex is simpler! In <em>ICML</em>, 2011.</li>
        <li id="BibPLXBIB0039" label="[39]">Z.&nbsp;Li, J.&nbsp;Wang, and J.&nbsp;Han. ePeriodicity: Mining event periodicity from incomplete observations. <em>TKDE</em>, 2015.</li>
        <li id="BibPLXBIB0040" label="[40]">S.&nbsp;Ma and J.&nbsp;L. Hellerstein. Mining partially periodic event patterns with unknown periods. In <em>International Conference on Data Engineering Proceedings</em>, 2001.</li>
        <li id="BibPLXBIB0041" label="[41]">B.&nbsp;Mc&nbsp;Mahon et&nbsp;al. Seasonal difference in brain serotonin transporter binding predicts symptom severity in patients with seasonal affective disorder. <em>Brain</em>, 2016.</li>
        <li id="BibPLXBIB0042" label="[42]">J.&nbsp;M. Mitchison. <em>The Biology of the Cell Cycle</em>. 1971.</li>
        <li id="BibPLXBIB0043" label="[43]">T.&nbsp;H. Monk, D.&nbsp;J. Buysse, L.&nbsp;R. Rose, J.&nbsp;A. Hall, and D.&nbsp;J. Kupfer. The sleep of healthy people – a diary study. <em>Chronobiol. Int</em>, 2000.</li>
        <li id="BibPLXBIB0044" label="[44]">S.&nbsp;A. Narod, H.&nbsp;Risch, R.&nbsp;Moslehi, A.&nbsp;Dørum, S.&nbsp;Neuhausen, H.&nbsp;Olsson, D.&nbsp;Provencher, P.&nbsp;Radice, G.&nbsp;Evans, S.&nbsp;Bishop, et&nbsp;al. Oral contraceptives and the risk of hereditary ovarian cancer. <em>New England Journal of Medicine</em>, 1998.</li>
        <li id="BibPLXBIB0045" label="[45]">K.&nbsp;A. Oinonen and D.&nbsp;Mazmanian. To what extent do oral contraceptives influence mood and affect? <em>Journal of Affective Disorders</em>, 2002.</li>
        <li id="BibPLXBIB0046" label="[46]">C.&nbsp;G. on&nbsp;Hormonal Factors&nbsp;in Breast&nbsp;Cancer et&nbsp;al. Breast cancer and hormonal contraceptives: collaborative reanalysis of individual data on 53,297 women with breast cancer and 100,239 women without breast cancer from 54 epidemiological studies. <em>The Lancet</em>, 1996.</li>
        <li id="BibPLXBIB0047" label="[47]">S.&nbsp;Orlando, P.&nbsp;Palmerini, R.&nbsp;Perego, and F.&nbsp;Silvestri. Adaptive and resource-aware mining of frequent sets. In <em>ICDM</em>, 2002.</li>
        <li id="BibPLXBIB0048" label="[48]">K.&nbsp;E. Paige. Effects of oral contraceptives on affective fluctuations associated with the menstrual cycle. <em>Psychosomatic Medicine</em>, 1971.</li>
        <li id="BibPLXBIB0049" label="[49]">T.&nbsp;Partonen and J.&nbsp;Lönnqvist. Seasonal affective disorder. <em>The Lancet</em>, 1998.</li>
        <li id="BibPLXBIB0050" label="[50]">T.&nbsp;Pearlstein, K.&nbsp;A. Yonkers, R.&nbsp;Fayyad, and J.&nbsp;A. Gillespie. Pretreatment pattern of symptom expression in premenstrual dysphoric disorder. <em>Journal of affective disorders</em>, 2005.</li>
        <li id="BibPLXBIB0051" label="[51]">R.&nbsp;S. Pindyck and D.&nbsp;L. Rubinfeld. <em>Econometric models and economic forecasts</em>. 1998.</li>
        <li id="BibPLXBIB0052" label="[52]">L.&nbsp;R. Rabiner. A tutorial on hidden Markov models and selected applications in speech recognition. <em>Proceedings of the IEEE</em>, 1989.</li>
        <li id="BibPLXBIB0053" label="[53]">T.&nbsp;Roenneberg, K.&nbsp;V. Allebrandt, M.&nbsp;Merrow, and C.&nbsp;Vetter. Social jetlag and obesity. <em>Current Biology</em>, 2012.</li>
        <li id="BibPLXBIB0054" label="[54]">J.&nbsp;Schreiber. Pomegranate: fast and flexible probabilistic modeling in Python. <em>arXiv preprint arXiv:1711.00137</em>, 2017.</li>
        <li id="BibPLXBIB0055" label="[55]">A.&nbsp;Shameli, T.&nbsp;Althoff, A.&nbsp;Saberi, and J.&nbsp;Leskovec. How gamification affects physical activity: Large-scale analysis of walking challenges in a mobile application. In <em>WWW</em>, 2017.</li>
        <li id="BibPLXBIB0056" label="[56]">P.&nbsp;Smyth. Clustering sequences with Hidden Markov Models. <em>NIPS</em>, 1997.</li>
        <li id="BibPLXBIB0057" label="[57]">R.&nbsp;Tibshirani, G.&nbsp;Walther, and T.&nbsp;Hastie. Estimating the number of clusters in a data set via the gap statistic. <em>JRSS-B</em>, 2001.</li>
        <li id="BibPLXBIB0058" label="[58]">X.&nbsp;Wang, D.&nbsp;Sontag, and F.&nbsp;Wang. Unsupervised learning of disease progression models. In <em>KDD</em>, 2014.</li>
        <li id="BibPLXBIB0059" label="[59]">E.&nbsp;M. Wojtys, L.&nbsp;J. Huston, T.&nbsp;N. Lindenfeld, T.&nbsp;E. Hewett, and M.&nbsp;L.&nbsp;V. Greenfield. Association between the menstrual cycle and anterior cruciate ligament injuries in female athletes. <em>The American Journal of Sports Medicine</em>, 1998.</li>
        <li id="BibPLXBIB0060" label="[60]">G.&nbsp;Wu, J.&nbsp;Zhu, J.&nbsp;Yu, L.&nbsp;Zhou, J.&nbsp;Z. Huang, and Z.&nbsp;Zhang. Evaluation of five methods for genome-wide circadian gene identification. <em>Journal of Biological Rhythms</em>, 2014.</li>
        <li id="BibPLXBIB0061" label="[61]">J.&nbsp;Yang, W.&nbsp;Wang, and P.&nbsp;S. Yu. Infominer: mining surprising periodic patterns. In <em>Knowledge Discovery and Data Mining</em>, 2001.</li>
        <li id="BibPLXBIB0062" label="[62]">J.&nbsp;Yang, W.&nbsp;Wang, and P.&nbsp;S. Yu. Mining asynchronous periodic patterns in time series data. <em>TKDE</em>, 2003.</li>
        <li id="BibPLXBIB0063" label="[63]">R.&nbsp;Yang and Z.&nbsp;Su. Analyzing circadian expression data by harmonic regression based on autoregressive spectral estimation. <em>Bioinformatics</em>, 2010.</li>
        <li id="BibPLXBIB0064" label="[64]">K.&nbsp;A. Yonkers, P.&nbsp;S. O'Brien, and E.&nbsp;Eriksson. Premenstrual syndrome. <em>The Lancet</em>, 2008.</li>
        <li id="BibPLXBIB0065" label="[65]">S.-Z. Yu. Hidden semi-Markov models. <em>Artificial Intelligence</em>, 2010.</li>
      </ul>
    </section>
  </section>
  <section id="foot-001" class="footnote">
    <header>
      <div class="title-info">
        <h2>FOOTNOTE</h2>
      </div>
    </header>
    <p id="fn1"><a href="#foot-fn1"><sup>1</sup></a>While in theory some duration distributions can allow an infinite number of substates for each state, in practice duration distributions have negligible mass beyond a certain maximum duration <em>d<sub>max</sub></em> , so the number of substates can be safely truncated.</p>
    <p id="fn2"><a href="#foot-fn2"><sup>2</sup></a>Categorical features could be modeled using a simple extension of our Bernoulli model; as there are no categorical features in our real-world datasets, we do not consider them here.</p>
    <p id="fn3"><a href="#foot-fn3"><sup>3</sup></a>To initialize the clustering, we choose a random subset of <span class="inline-equation"><span class="tex">$\mathcal {S}$</span></span> individuals and fit one CyHMM <span class="inline-equation"><span class="tex">$\mathcal {M}_s$</span></span> for each individual. We then loop over all individuals in the dataset and compute the log likelihood <span class="inline-equation"><span class="tex">$\mathcal {L}_{s}^{(i)}$</span></span> for each individual <em>i</em> under each model <span class="inline-equation"><span class="tex">$\mathcal {M}_s$</span></span> . We run k-means on the matrix of these likelihoods to initially divide individuals into <em>C</em> clusters (z-scoring each individual to control for the fact that, for example, individual time series vary in length and thus log likelihoods may not be directly comparable). This initialization procedure is similar to [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top" href="#BibPLXBIB0056">56</a>] but their procedure is quadratic in the number of individuals, rendering it too slow for large real-world datasets (Section <a class="sec" href="#sec-15">6</a>).</p>
    <p id="fn4"><a href="#foot-fn4"><sup>4</sup></a><a href="https://github.com/jcborges/PeriodicEventMining" target="_blank">https://github.com/jcborges/PeriodicEventMining</a></p>
    <p id="fn5"><a href="#foot-fn5"><sup>5</sup></a><a href="https://github.com/cauyrd/ARSER" target="_blank">https://github.com/cauyrd/ARSER</a></p>
    <p id="fn6"><a href="#foot-fn6"><sup>6</sup></a>Data from both companies has been used in previous studies and users are informed that their data may be used. Because all data analyzed is preexisting and de-identified, the analysis is exempt from IRB review.</p>
    <p id="fn7"><a href="#foot-fn7"><sup>7</sup></a>We report results using the 4-state CyHMM, but also found that 3 and 5 state CyHMMs significantly outperformed baselines. We also compared to the baseline of a CyHMM with a geometric duration distribution (rather than a Poisson duration distribution). The errors in inferred cycle length using a geometric rather than a Poisson distribution were 10%, 47%, and 129% higher for the 3, 4, and 5 state models, respectively, confirming that the Poisson distribution provided a better fit to the data.</p>
    <div class="bibStrip">
      <p>This paper is published under the Creative Commons Attribution 4.0 International (CC-BY&nbsp;4.0) license. Authors reserve their rights to disseminate the work on their personal and corporate Web sites with the appropriate attribution.</p>
      <p><em>WWW '18, April 23-27, 2018, Lyon, France</em></p>
      <p>© 2018; IW3C2 (International World Wide Web Conference Committee), published under Creative Commons CC-BY&nbsp;4.0 License. ACM ISBN 978-1-4503-5639-8/18/04.<br />
      DOI: <a class="link-inline force-break" target="_blank" href="https://doi.org/10.1145/3178876.3186052">https://doi.org/10.1145/3178876.3186052</a></p>
    </div>
  </section>
  <div class="pubHistory">
    <p></p>
  </div>
</body>
</html>
