<!DOCTYPE html> <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">  <head>  <title>Presenting and Preserving the Change in Taxonomic Knowledge for Linked Data*</title>  <!-- Copyright (c) 2010-2015 The MathJax Consortium --><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"></meta>  <meta name="viewport" content="width=device-width; initial-scale=1.0;"></meta>  <meta http-equiv="X-UA-Compatible" content="IE=edge"></meta>  <link media="screen, print" rel="stylesheet"    href="../../../data/dl.acm.org/pubs/lib/css/bootstrap.min.css"/><link media="screen, print" rel="stylesheet"    href="../../../data/dl.acm.org/pubs/lib/css/bootstrap-theme.min.css"/><link media="screen, print" rel="stylesheet"    href="../../../data/dl.acm.org/pubs/lib/css/main.css"/><script src="../../../data/dl.acm.org/pubs/lib/js/jquery.min.js" type="text/javascript"></script>  <script src="../../../data/dl.acm.org/pubs/lib/js/bootstrap.min.js" type="text/javascript"></script>  <script src="../../../data/dl.acm.org/pubs/lib/js/bibCit.js" type="text/javascript"></script>  <script src="../../../data/dl.acm.org/pubs/lib/js/divTab.js" type="text/javascript"></script>  <script type="text/javascript"    src="../../../data/dl.acm.org/pubs/lib/js/MathJax.TeX-AMS_CHTML.js"></script>  <script type="text/x-mathjax-config">MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}});</script>  </head>  <body id="main">  <section class="front-matter">   <section>    <header class="title-info">     <div class="journal-title">     <h1>      <span class="title">Presenting and Preserving the Change in Taxonomic Knowledge for Linked Data*</span>      <br/>      <span class="subTitle"/>     </h1>     </div>    </header>    <div class="authorGroup">     <div class="author">     <span class="givenName">Rathachai</span>      <span class="surName">Chawuthai</span>     King Mongkut&#x0027;s Institute of Technology Ladkrabang, Thailand, <a href="mailto:rathachai.ch@kmitl.ac.th">rathachai.ch@kmitl.ac.th</a>         </div>         <div class="author">     <span class="givenName">Hideaki</span>      <span class="surName">Takeda</span>     National Institute of Informatics, Japan, <a href="mailto:takeda@nii.ac.jp">takeda@nii.ac.jp</a>         </div>         <div class="author">     <span class="givenName">Vilas</span>      <span class="surName">Wuwongse</span>     Mahidol Univeristy, Thailand, <a href="mailto:vilas.wuw@mahidol.ac.th">vilas.wuw@mahidol.ac.th</a>         </div>         <div class="author">     <span class="givenName">Utsugi</span>      <span class="surName">Jinbo</span>     National Museum of Nature and Science, Japan, <a href="mailto:ujinbo@kahaku.go.jp">ujinbo@kahaku.go.jp</a>         </div>        </div>    <br/>    <div class="pubInfo">     <p>DOI: <a href="https://doi.org/10.1145/3184558.3186234" target="_blank">https://doi.org/10.1145/3184558.3186234</a>     <br/>WWW '18: <a href="https://doi.org/10.1145/3184558" target="_blank">Proceedings of The Web Conference 2018</a>, Lyon, France, April 2018</p>    </div>    <div class="abstract">     <p>     <small>Linked Open Data (LOD) technology enables a web of data and exchangeable knowledge graphs through the Internet. However, the change in knowledge is happened everywhere and every time. This issue is commonly found in every domain especially in taxonomic knowledge. In fact, biological classification and taxonomic names are not consistent among different databases due to new discovery and various viewpoints of taxonomists. This issue leads to ambiguity in taxon interpretation and imprecise linked data. In this case, the temporal representation of taxa and underlying knowledge of the change in taxonomy are considered to integrate with taxonomic data models as well. For this reason, this work objects to introduce an approach to the preservation and the presentation of change in taxonomic knowledge, and to produce linked data for supporting world-wide knowledge exchange. The main outcome is an ontology including operations for presenting changes, context-based human-readable identifiers indicating revision of taxon concepts, an event-based data model for preserving changes with contextual information, and two linked data models for presenting chronological changes in taxa and their temporal information at a given time point. All of these features are originated to provide a better understanding of organisms to learners. In addition, a prototype is implemented to demonstrate the feasibility of the proposed approach against the real cases from domain experts. The result shows that our approach can support various practical cases of changes in taxonomic knowledge and provides open and accurate access to LOD cloud.</small>     </p>    </div>    <div class="classifications">     <div class="author">     <span style="font-weight:bold;">      <small>KEYWORDS:</small>     </span>     <span class="keyword">      <small>Biodiversity Informatics</small>, </span>     <span class="keyword">      <small>Change in Taxonomy</small>, </span>     <span class="keyword">      <small>Knowledge Representation</small>, </span>     <span class="keyword">      <small>Knowledge Exchange</small>, </span>     <span class="keyword">      <small>Linked Data</small>, </span>     <span class="keyword">      <small>Ontology</small>, </span>     <span class="keyword">      <small>RDF</small>, </span>     <span class="keyword">      <small>Semantic Web</small>, </span>     <span class="keyword">      <small>Taxonomic Data</small>     </span>     </div>     <br/>     <div class="AcmReferenceFormat">     <p>      <small>       <span style="font-weight:bold;">ACM Reference Format:</span>       <br/>Rathachai Chawuthai, Hideaki Takeda, Vilas Wuwongse and Utsugi Jinbo. 2018. Presenting and Preserving the Change in Taxonomic Knowledge for Linked Data. In <em>Proceedings of The 2018 Web Conference Companion (WWW'18 Companion). ACM, New York, NY, USA, 7 pages.</em> <a href="https://doi.org/10.1145/3184558.3186234" class="link-inline force-break"        target="_blank">https://doi.org/10.1145/3184558.3186234</a></small>     </p>     </div>    </div>   </section>  </section>  <section class="body">   <section id="sec1">    <header>     <div class="title-info">     <h2>      <span class="section-number">1</span> INTRODUCTION</h2>     </div>    </header>    <p>Linked Open Data (LOD) technology attempts to transform web of documents into web of data and make them become exchangeable through the Internet [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib1">1</a>]. Thus, all pieces of knowledge around the world becomes in form of semantic knowledge graphs which are mathematically described by a binary relation, and the advantage of this format is to have data become easily linkable [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib2">2</a>]. However, knowledge is generally dynamic and subjected to change over time due to new discovery and new justification from humans. A lot of out-of-date and up-to-date pieces of knowledge are intermixed everywhere among sources of knowledge in the Internet. This issue is one cause of the misinterpretation and misunderstanding of some terms or some concepts from chronological transaction records under the different context of time and across communities, and it results in the inaccuracy of linked data. Integrating an aspect of time in a knowledge graph becomes an appropriate solution, but it cannot be done straightforwardly due to the limitation of the binary relation. For this reason, our work efforts to introduce an approach to preserving and presenting the change in knowledge graphs for linked data, and we take this approach into a biodiversity domain because taxonomic knowledge under this domain is commonly changed and has clearly chronological transaction records documented.</p>    <p>Taxonomy (or biological taxonomy) is the sciences of naming and classifying the group of organisms considered by their characteristic shared. Each group in the hierarchical classification is called &#x201C;taxon&#x201D; (plural &#x201C;taxa&#x201D;) and it must be named. A lot of articles about taxonomic knowledge have been published around the world for 200 years. These pieces of knowledge could be linked for increasing learning capacity. However, there is no precise referenceable key to associate these pieces of knowledge because there is no globally accepted scientific names and classifications [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib3">3</a>]. There are more than one scientific names for a species, and there are many changes in the classifications [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib4">4</a>-<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib7">7</a>].</p>    <p>For example, two genera of snowy owls, <em>Bubo</em> and <em>Nyctea</em>, have been merged under the name <em>Bubo</em> since 1999 [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib5">5</a>]. According to the zoological nomenclature rule [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib8">8</a>], the species <em>Nyctea scandiaca</em>, which was under the genus <em>Nyctea</em>, have been transferred into the newly accepted genus <em>Bubo</em> with the new name <em>Bubo scandiacus</em> [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib5">5</a>]. There are a lot of online documents about this snowy owl but there are detailed with these two names separately. For having knowledge exchange, changes in taxonomy of these species must be records with precise context in a standard data model for linked data.</p>    <p>For this reason, we conduct this research whose objectives are to: (1) preserve the change in taxonomic knowledge, and (2) present and publish taxonomic knowledge as linked data.</p>    <p>As we studied from other taxonomic databases [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib9">9</a>-<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib17">17</a>], the online linked taxonomic knowledge is needed, and those works do not support our research objectives.</p>    <p>To accomplish these two objectives, we implement an approach named Linked Taxonomic Knowledge (LTK) based on the idea of two previous works, Contextual Knowledge for Archives (CKA) approach [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib18">18</a>] and the Meta-Ontology of Biological Name (TaxMeOn) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib19">19</a>]. We also reuse some taxonomic terms from Linked Open Data for ACademia (LODAC) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib20">20</a>] and Simple Knowledge Organization System (SKOS) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib21">21</a>] to manage the relationships between concepts, and publish data to the Linked Open Data (LOD) Cloud [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib1">1</a>]. The LTK and its prototype will be described hereafter.</p>   </section>   <section id="sec2">    <header>     <div class="title-info">     <h2>      <span class="section-number">2</span> A LOGICAL MODEL FOR LINKING TAXONOMIC KNOWLEDGE</h2>     </div>    </header>    <p>The model of LTK is proposed based two points: (1) the model can preserve the changes as an event along with aspects of time and provenance, and allows tracing background knowledge behind the change; and (2) the model can make linked data be human- and machine-readable, and support the chronological and temporal data presentation.</p>    <p>All data models are designed to be compatible with Semantic web technology, so all of them can be represented by the Resource Description Framework (RDF) format [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib22">22</a>]. RDF data are in the form of a subject&#x2013;predicate&#x2013;object expression, known as a triple, and many triples generate an RDF graph which becomes knowledge graph as well.</p>    <p>The detail our work is illustrated as follows:</p>    <section id="sec2Z1">     <header>     <div class="title-info">      <h3>       <span class="section-number">2.1</span> Structure to Represent Change in Taxonomic Knowledge</h3>     </div>     </header>     <p>On the basis of these changes analyzed from actual use cases [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#bib9">9</a>-<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#bib17">17</a>], we first categorize changes in taxonomic knowledge into three mains types as shown in <a class="fig" href="#fig1">Fig. 1</a>. Change in nomenclature is the change in the name of taxa, change in a taxon concept is the change in circumscription or classification scope of a taxon, and change in relationship is change in association or link between two taxon concepts. There are about 15 terms in the leaf nodes that will become predefined operations representing changes under our LTK ontology.</p>     <figure id="fig1">     <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186234/images/image1.png" class="img-responsive" alt="Figure 1:" longdesc=""/>     <div class="figure-caption">      <span class="figure-number">Figure 1:</span>      <span class="figure-title">Overall data models of LTK</span>     </div>     </figure>     <p>In practice, some following operations are often used. <em>Synonym</em> is a link between different names of same taxon. <em>Merging</em> is to lump two or more taxa into one taxon. <em>Splitting</em> is to separate one taxon into two or more taxa. <em>Changing higher taxon</em> is to transfer a lower taxon to another super taxon. For merging and splitting, the names of taxa before the change become obsolete but not the selected names after the change.</p>    </section>    <section id="sec2Z2">     <header>     <div class="title-info">      <h3>       <span class="section-number">2.2</span> Entities for LTK</h3>     </div>     </header>     <p>An entity is a set of names represented by Uniform Resource Identifier (URI). For example, the URI of the genus Bubo is <em>http://rc.lodac.nii.ac.jp/taxon/genus/Bubo</em>; and it can be shorten to <em>genus:Bubo</em> according to user-defined prefixes. There are three types of entities.</p>     <p>     <strong>Nominal Entity</strong> (<strong>      <em>ltk:NOM</em>     </strong>) is a set of URIs represented taxonomic concepts (taxon concept) which can be either names or identifiers used in a taxonomic information system.</p>     <p>     <strong>Simple Nominal Entity</strong> (<strong>      <em>ltk:SIM</em>     </strong>) is a subset of <em>ltk:NOM</em> which contains only human-readable URIs such as <em>genus:Bubo</em>.</p>     <p>     <strong>Contextual Nominal Entity</strong> (<strong>      <em>ltk:CON</em>     </strong>) is a subset of <em>ltk:NOM</em> which presents a version number in a URI. For example, <em>genus:Bubo_1999</em> has been a version of <em>Bubo</em> since the year 1999. This entity is generally used in the LTK ontology, and it always links to the <em>ltk:SIM</em> for linking to an LOD cloud.</p>    </section>    <section id="sec2Z3">     <header>     <div class="title-info">      <h3>       <span class="section-number">2.3</span> Operations representing Changes</h3>     </div>     </header>     <p>According to Section <a class="sec" href="#sec2Z1">2.1</a>, the structure of changes is used to be operations for presenting the change in data models.</p>     <p>     <strong>Operation of Change</strong> (<strong>      <em>ltk:OPR</em>     </strong>) is a set of types of changes used in the LTK ontology. It includes the followings subtypes.</p>     <p>     <strong>Operation of Change in Conception</strong> (<strong>      <em>ltk:OPRC</em>     </strong>) is a set of the changes in the circumscription (scope) of taxon concepts such as splitting a taxon into taxa, merging taxa in to a taxon, and replacing a taxon into another one. The URIs for them are <em>ltk:TaxonMerger, ltk:TaxonSplitter</em>, and <em>ltk:TaxonReplacement</em>.</p>     <p>     <strong>Operation of Change in Relation</strong> (<strong>      <em>ltk:OPRR</em>     </strong>) is a set of the changes in relations or associations between two taxon concepts, such as changing a higher taxon to another one, subdividing a taxon into lower taxa, combining taxa into a higher taxon, and linking a taxon to its synonym. The URIs for them are <em>ltk:ChangeHigherTaxon, ltk:SubdivideTaxon, ltk:CombineTaxa</em>, and <em>ltk:SynonymLink</em>.</p>    </section>    <section id="sec2Z4">     <header>     <div class="title-info">      <h3>       <span class="section-number">2.4</span> Data Models for LTK</h3>     </div>     </header>     <p>There are three data models. We first introduce an Event-Centric model that preserve any changes in taxonomy. This model contains aspects of time and provenance data that uses the idea of <em>n</em>-ary relation but presented by the binary relation, so the model becomes complicated by design and it is not suitable for linking to LOD. Thus, we next create two data views of the Event-Centric model; there are a Transition model and a Snapshot model in order make light-weight data models for exchanging knowledge through LOD cloud as shown in <a class="fig" href="#fig2">Fig. 2.</a>     </p>     <figure id="fig2">     <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186234/images/image2.png" class="img-responsive" alt="Figure 2:" longdesc=""/>     <div class="figure-caption">      <span class="figure-number">Figure 2:</span>      <span class="figure-title">Overall data models of LTK</span>     </div>     </figure>    </section>    <section id="sec2Z5">     <header>     <div class="title-info">      <h3>       <span class="section-number">2.5</span> Event-Centric Model</h3>     </div>     </header>     <p>In detail, the Event-Centric model assures <em>Operations</em> of change (by a property <em>cka:assure</em>). It also includes provenance data to show the <em>Interval</em> time of that change (by a property <em>cka:interval</em>) with have beginning time (<em>tl:begin</em>) and end time (<em>tl:end</em>), to inform <em>Persons</em> who perform or issue this change (by properties <em>bibo:Performer</em> and <em>bibo:issuer</em>), to show relationships and consequences of the change (by properties <em>cka:main, cka:detail, cka:cause,</em> and <em>cka:effect</em>), and to cite some referent <em>Sources</em> (by a property <em>dct:source</em>) as shown in <a class="fig" href="#fig3">Fig. 3.</a>     </p>     <figure id="fig3">     <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186234/images/image3.png" class="img-responsive" alt="Figure 3:" longdesc=""/>     <div class="figure-caption">      <span class="figure-number">Figure 3:</span>      <span class="figure-title">Event-Centric Model Schema</span>     </div>     </figure>     <p>For example, the case of change in the taxonomy of snowy owls [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#bib5">5</a>,<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#bib8">8</a>] can be represented by the following RDF statements.</p>     <div class="table-responsive" id="Utb1">     <table class="table">      <tbody>       <tr>        <td style="text-align:right;">1</td>        <td style="text-align:left;">ex:event1999</td>       </tr>       <tr>        <td style="text-align:right;">2</td>        <td style="text-align:left;">bibo:issuer	pp:Richard ;</td>       </tr>       <tr>        <td style="text-align:right;">3</td>        <td style="text-align:left;">cka:interval	[tl:begin &#x201C;1999&#x201D;];</td>       </tr>       <tr>        <td style="text-align:right;">4</td>        <td style="text-align:left;">cka:assures 	ex:mg1, ex:rp1 .</td>       </tr>       <tr>        <td style="text-align:right;">5</td>        <td style="text-align:left;"/>       </tr>       <tr>        <td style="text-align:right;">6</td>        <td style="text-align:left;">ex:mg1	rdf:type ltk:TaxonMerger ;</td>       </tr>       <tr>        <td style="text-align:right;">7</td>        <td style="text-align:left;">ltk:taxonBefore genus:Bubo_1805 ;</td>       </tr>       <tr>        <td style="text-align:right;">8</td>        <td style="text-align:left;">ltk:taxonBefore genus:Nyctea_1826 ;</td>       </tr>       <tr>        <td style="text-align:right;">9</td>        <td style="text-align:left;">ltk:taxontAfter genus:Bubo_1999 .</td>       </tr>       <tr>        <td style="text-align:right;">10</td>        <td style="text-align:left;"/>       </tr>       <tr>        <td style="text-align:right;">11</td>        <td style="text-align:left;">ex:syn1	rdf:type ltk:SynonymLink;</td>       </tr>       <tr>        <td style="text-align:right;">12</td>        <td style="text-align:left;">ltk:sourceTaxon</td>       </tr>       <tr>        <td style="text-align:right;">13</td>        <td style="text-align:left;">species:Nyctea_scandiaca_1826 ;</td>       </tr>       <tr>        <td style="text-align:right;">14</td>        <td style="text-align:left;">ltk:targetTaxon</td>       </tr>       <tr>        <td style="text-align:right;">15</td>        <td style="text-align:left;">species:Bubo_scandiacus_1999 .</td>       </tr>       <tr>        <td style="text-align:right;">16</td>        <td style="text-align:left;"/>       </tr>       <tr>        <td style="text-align:right;">17</td>        <td style="text-align:left;">ex:mg1 cka:effect ex:syn1 .</td>       </tr>      </tbody>     </table>     </div>     <p>We use contextual nominal entities for presenting scientific names because non-semantic-web-expert users especially in biological researchers can use less effort on recognizing it. All URIs in the RDF statements are shorten using user-defined prefixes. Lines 1-4 show that the event entity refers to some operations and provenance data. Lines 6-9 show the merging of two genera into the <em>genus:Bubo_1999</em>. Lines 11-15 express the synonym of both species. Line 17 presents that the merging of two genera contributes to the synonym of both species.</p>     <p>The <em>n</em>-ary data structure of the Event-Centric model is like a data model in a relational database. It is suitable to embed contextual information of change for preserving in a knowledge base and become background knowledge of every change. However, using the binary relation for presenting the <em>n</em>-ary relation makes the data model be complex, so its design is not proper for consuming by any LOD applications directly. In this case, the Event-Centric model must be transformed into general knowledge graphs that are the Transition model and the Snapshot model for publishing data to the LOD cloud.</p>    </section>    <section id="sec2Z6">     <header>     <div class="title-info">      <h3>       <span class="section-number">2.6</span> Transition Model</h3>     </div>     </header>     <p>The Transition model is a knowledge graph for presenting the chronological chain of the changes in contextual nominal entities. The model is simpler than the Event-Centric model because its contextual information is omitted. For example, the RDF statements presenting the Event-Centric model of the change in taxonomy snowy owls from Section <a class="sec" href="#sec2Z5">2.5</a> can be transformed into a simple knowledge graph as shown in <a class="fig" href="#fig4">Fig. 4.</a>     </p>     <figure id="fig4">     <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186234/images/image4.png" class="img-responsive" alt="Figure 4:" longdesc=""/>     <div class="figure-caption">      <span class="figure-number">Figure 4:</span>      <span class="figure-title">Example data of the Transition Model</span>     </div>     </figure>     <p>According to that figure, the property <em>ltk:mergedInto</em> is appeared to be a link between taxa before and after merging. This property is a linking property and it is bound with the operation <em>ltk:TaxonMerger</em> as expressed in the following RDF statements. Our approach restricts that every operation must be declared in the LTK ontology.</p>     <div class="table-responsive" id="Utb2">     <table class="table">      <tbody>       <tr>        <td style="text-align:right;">1<br/>2</td>        <td style="text-align:left;">ltk:TaxonMerger	<br/>ltk:linkingProperty ltk:mergedInto .</td>       </tr>      </tbody>     </table>     </div>     <p>This declaration supports the transformation rules to convert the Event-Centric model into the Transition model. The rules are well-defined based on the Semantic Web Rule Language, and it is executed by Semantic Web reasoning engine such as Jena [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#bib23">23</a>].</p>     <p>Properties in the Transition model are link to some properties from well-known ontology for example, <em>ltk:mergedInto</em> is a sub-property of <em>skos:broadMatch</em>. If users make a query with the property <em>skos:broadMatch</em>, they will get the result from the property <em>ltk:mergedInto</em> too. It also can trace back to data in the Even-Centric model to view reason behind a particular transition pair. Thus, it becomes advantage to publish the chronological changes in taxa to LOD cloud.</p>    </section>    <section id="sec2Z7">     <header>     <div class="title-info">      <h3>       <span class="section-number">2.7</span> Snapshot Model</h3>     </div>     </header>     <p>As well as the Transition model, the Snapshot model is designed to be light-weight triples presenting temporal RDF data at a specific time point. For example, if the species <em>Nyctea scandica</em> is cited before 1999, it is considered as an accepted species. However, when it is mentioned after 1999, it is a synonymy of the <em>Bubo scandiacus</em> as shown in <a class="fig" href="#fig5">Fig. 5</a>. It means that when we learn about <em>Bubo scandiacus</em>, we should link to any information related to <em>Nyctea scandica</em> documented after 1999 as well.</p>     <figure id="fig5">     <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186234/images/image5.png" class="img-responsive" alt="Figure 5:" longdesc=""/>     <div class="figure-caption">      <span class="figure-number">Figure 5:</span>      <span class="figure-title">Example data of the Snapshot Model since 1999</span>     </div>     </figure>     <p>The implementation of the transformation process is similar to the Transition model. We firstly declare this operation with a relation property. For example, the relation <em>ltk:synonym</em> is bounded with the operation <em>ltk:SynonymLink</em>. Then, a proposed rule is able to convert the Event-Centric model into the Snapshot model presenting temporal information of taxa based on a given time point.</p>    </section>   </section>   <section id="sec3">    <header>     <div class="title-info">     <h2>      <span class="section-number">3</span> APPLICATION</h2>     </div>    </header>    <p>To test the feasibility of our approach, a web application is implemented. As shown in <a class="fig" href="#fig6">Fig. 6.</a>, the application has three layers: the first layer contains a web application, web services and a query endpoint; the second layer has a rule engine and ontologies; and the last layer provides RDF data. Thus, users can access the web application at <em>http://rc.lodac.nii.ac.jp/ltk/</em>, while other client applications or services can access the system through the web services and the query endpoint. The system also provides dereferenceable URIs that returns a document type according to the mine type in the request header.</p>    <figure id="fig6">     <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186234/images/image6.png" class="img-responsive" alt="Figure 6:" longdesc=""/>     <div class="figure-caption">     <span class="figure-number">Figure 6:</span>     <span class="figure-title">Software architecture of the prototype</span>     </div>    </figure>    <p>We also test our approach with real test cases of Japanese moths under the family Saturniidae published as three checklists (list of names): Inoue in 1982 [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib24">24</a>], Jinbo in 2008 [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib25">25</a>], and Kishida in 2011 [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#bib26">26</a>]. It has been found that our work can handle all the cases and all data are linked with any Internet resources from well-known ontologies in the LOD cloud. The domain experts from biodiversity informatics and Semantic Web accept that this work gives contribution to the biodiversity domain in terms of knowledge representation, user engagement, and system integration; and it can be applied to any other domains as well.</p>    <section id="sec4">     <header>     <div class="title-info">      <h3>       <span class="section-number">4</span> CONCLUSION</h3>     </div>     </header>     <p>Our work introduced an approach to the preservation and the presentation of changes in knowledge for linked data by focusing on the biodiversity domain. Changes in taxonomic classification and names are the key issue of our research. The approach contains the LTK ontology; and it includes the Entities of concept, the Operations for categorizing changes, and the Data models for documenting changes. The primary data model is the Event-Centric model that captures the change with aspects of time and provenance. The secondary data models that are the Transition model and the Snapshot model are initiates to be the light-weight views of the Event-Centric model. The simplicity of the last two data models are proper for being used by LOD applications. In addition, the prototype has been developed to tested with the real cases of changes in taxonomic knowledge. The outcome indicates that our work is one contribution to empower the capability of knowledge management for global accessibility across communities and time using linked data.</p>    </section>   </section>  </section>  <section class="back-matter">   <section id="bib-sec-001">    <header>     <div class="title-info">     <h2 class="page-brake-head">REFERENCES</h2>     </div>    </header>    <ul class="bibUl">     <li id="bib1" label="[1]">T. Heath and B. Christian, &#x201C;Linked data: Evolving the web into a global data space,&#x201D; in <em>Synthesis lectures on the semantic web: theory and technology</em>, 2011.</li>     <li id="bib2" label="[2]">P. Hitzler, M. Krotzsch and S. Rudolph, Foundations of semantic web technologies, CRC Press, 2009.</li>     <li id="bib3" label="[3]">N. Franz and R. Peet, &#x201C;Perspectives: towards a language for mapping relationships among taxonomic concepts,&#x201D; <em>Systematics and Biodiversity,</em> vol. 7, no. 1, pp. 5-20, 2009.</li>     <li id="bib4" label="[4]">J. E. Winston, Describing species: practical taxonomic procedure for biologists, Columbia University Press, 1999.</li>     <li id="bib5" label="[5]">International Commission on Zoological Nomenclature: International Code of Zoological Nomenclature, 4 ed., International Trust for Zoological Nomenclatu History Museum, 1999.</li>     <li id="bib6" label="[6]">C. G. Sibley and L. L. Short, &#x201C;Hybridization in the orioles of the Great Plains,&#x201D; <em>Condor,</em> pp. 130-150, 1964.</li>     <li id="bib7" label="[7]">S. Freeman and R. M. Zink, &#x201C;A phylogenetic study of the blackbirds based on variation in mitochondrial DNA restriction sites,&#x201D; <em>Systematic Biology,</em> vol. 44, no. 3, pp. 409-420, 1995.</li>     <li id="bib8" label="[8]">M. Wink and P. Heidrich, &#x201C;Molecular evolution and systematics of owls (Strigiformes),&#x201D; <em>Owls A Guide to the Owls of the World,</em> pp. 39-57, 1999.</li>     <li id="bib9" label="[9]">W. G. Berendsohn, &#x201C;A taxonomic information model for botanical databases: the IOPI model,&#x201D; <em>Taxon,</em> pp. 283-309, 1997.</li>     <li id="bib10" label="[10]">A. C. Jones, R. J. White and E. R. Orme, &#x201C;Identifying and relating biological concepts in the Catalogue of Life,&#x201D; <em>Journal of Biomedical Semantics,</em> vol. 2, no. 1, 2011.</li>     <li id="bib11" label="[11]">J. B. Kennedy, R. Kukla and T. Paterson, &#x201C;Scientific names are ambiguous as identifiers for biological taxa: their context and definition are required for accurate data integration,&#x201D; in <em>Data integration in the life sciences</em>, Springer, 2005, pp. 80-95.</li>     <li id="bib12" label="[12]">N. Laurenne, J. Tuominen, H. Saarenmaa and E. Hyvonen, &#x201C;Making species checklists understandable to machines&#x2013;a shift from relational databases to ontologies,&#x201D; <em>Journal of Biomedical Semantics,</em> vol. 5, no. 1, 2014.</li>     <li id="bib13" label="[13]">R. D. Page, &#x201C;Taxonomic names, metadata, and the Semantic Web,&#x201D; <em>Biodiversity Informatics,</em> vol. 3, 2006.</li>     <li id="bib14" label="[14]">D. J. Patterson, J. Cooper, P. M. Kirk and <em>et al.</em>, &#x201C;Names are key to the big new biology,&#x201D; <em>Trends in ecology &amp; evolution,</em> vol. 25, no. 12, pp. 686-691, 2010.</li>     <li id="bib15" label="[15]">I. N. Sarkar, &#x201C;Biodiversity informatics: organizing and linking information across the spectrum of life,&#x201D; <em>Briefings in Bioinformatics,</em> vol. 8, no. 5, pp. 347-357, 2007.</li>     <li id="bib16" label="[16]">S. Schulz, H. Stenzhorn and M. Boeker, &#x201C;The ontology of biological taxa,&#x201D; <em>Bioinformatics,</em> vol. 24, no. 13, pp. i313-i321, 2008.</li>     <li id="bib17" label="[17]">N. Ytow, D. R. Morse and D. M. Roberts, &#x201C;Nomencurator: a nomenclatural history model to handle multiple taxonomic views,&#x201D; <em>Biological journal of the Linnean Society,</em> vol. 73, no. 1, pp. 81-98, 2001.</li>     <li id="bib18" label="[18]">R. Chawuthai, V. Wuwongse and H. Takeda, &#x201C;A Formal Approach to the Modelling of Digital Archives,&#x201D; in <em>The Outreach of Digital Libraries: A Globalized Resource Network</em>, Springer, 2012, pp. 179-188.</li>     <li id="bib19" label="[19]">J. Tuominen, N. Laurenne and E. Hyvonen, &#x201C;Biological names and taxonomies on the semantic web: managing the change in scientific conception,&#x201D; in <em>The Semanic Web: Research and Applications</em>, Springer, 2011, pp. 255-269.</li>     <li id="bib20" label="[20]">Y. Minami, H. Takeda, F. Kato and <em>et al.</em>, &#x201C;Towards a Data Hub for Biodiversity with LOD,&#x201D; in <em>The 2nd Joint International Semantic Technology Conference</em>, 2013.</li>     <li id="bib21" label="[21]">&#x201C;Simple Knowledge Organization System,&#x201D; [Online]. Available: <a class="link-inline force-break" href="http://www.w3.org/TR/skos-primer/">http://www.w3.org/TR/skos-primer/</a>.</li>     <li id="bib22" label="[22]">G. Schreiber and Y. Raimond, &#x201C;RDF 1.1 Primer,&#x201D; 2014. [Online]. Available: <a class="link-inline force-break" href="https://www.w3.org/TR/rdf11-primer/">https://www.w3.org/TR/rdf11-primer/</a>.</li>     <li id="bib23" label="[23]">&#x201C;Apache Jena,&#x201D; [Online]. Available: <a class="link-inline force-break" href="http://jena.apache.org/">http://jena.apache.org/</a>.</li>     <li id="bib24" label="[24]">H. Inoue, S. Sugi, H. Kuroko and <em>et al.</em>, Moths of Japan, vol. 2. Plates and synonymic catalogue, vol. 2, Tokyo: Kodansha Tokyo, 1982.</li>     <li id="bib25" label="[25]">U. Jinbo, &#x201C;List-MJ: A checklist of Japanese moths 2004-2008,&#x201D; 2008. [Online]. Available: <a class="link-inline force-break" href="http://listmj.mothprog.com">http://listmj.mothprog.com</a>.</li>     <li id="bib26" label="[26]">Y. Kishida, The Standard of Moths in Japan II, vol. 2, Tokyo: Tokyo: Gakken, 2011.</li>    </ul>   </section>  </section>  <section id="foot-001" class="footnote">   <header>    <div class="title-info">     <h2>FOOTNOTE</h2>    </div>   </header>   <div class="bibStrip">    <p>This paper is published under the Creative Commons Attribution 4.0 International (CC-BY 4.0) license. Authors reserve their rights to disseminate the work on their personal and corporate Web sites with the appropriate attribution.</p>    <p>     <em>WWW '18, April 23-27, 2018, Lyon, France</em>    </p>    <p>&#x00A9; 2018 IW3C2 (International World Wide Web Conference Committee), published under Creative Commons CC BY 4.0 License. ACM ISBN 978-1-4503-5640-4/18/04.<br/>DOI: <a class="link-inline force-break" target="_blank"     href="https://doi.org/10.1145/3184558.3186234">https://doi.org/10.1145/3184558.3186234</a>    </p>   </div>  </section>  <div class="pubHistory">   <p/>  </div>  </body> </html> 
