<!DOCTYPE html> <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en"> <head>  <title>Mining the Web of Life Sciences Linked Open Data for Mechanism-Based Pharmacovigilance</title>  <!-- Copyright (c) 2010-2015 The MathJax Consortium --><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"></meta>  <meta name="viewport" content="width=device-width; initial-scale=1.0;"></meta>  <meta http-equiv="X-UA-Compatible" content="IE=edge"></meta>  <link media="screen, print" rel="stylesheet"    href="../../../data/dl.acm.org/pubs/lib/css/bootstrap.min.css"/><link media="screen, print" rel="stylesheet"    href="../../../data/dl.acm.org/pubs/lib/css/bootstrap-theme.min.css"/><link media="screen, print" rel="stylesheet"    href="../../../data/dl.acm.org/pubs/lib/css/main.css"/><script src="../../../data/dl.acm.org/pubs/lib/js/jquery.min.js" type="text/javascript"></script>  <script src="../../../data/dl.acm.org/pubs/lib/js/bootstrap.min.js" type="text/javascript"></script>  <script src="../../../data/dl.acm.org/pubs/lib/js/bibCit.js" type="text/javascript"></script>  <script src="../../../data/dl.acm.org/pubs/lib/js/divTab.js" type="text/javascript"></script>  <script type="text/javascript"    src="../../../data/dl.acm.org/pubs/lib/js/MathJax.js?config=TeX-AMS_CHTML"></script>  <script type="text/x-mathjax-config">MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}});</script> </head> <body id="main">  <section class="front-matter">   <section>    <header class="title-info">    <div class="journal-title">     <h1>      <span class="title">Mining the Web of Life Sciences Linked Open Data for Mechanism-Based Pharmacovigilance</span>      <br/>      <span class="subTitle"/>     </h1>    </div>    </header>    <div class="authorGroup">    <div class="author">     <span class="givenName">Maulik R.</span>     <span class="surName">Kamdar</span>,     Center for Biomedical Informatics Research Stanford 		 University &laquo;supervised by Mark A. Musen&raquo;, <a href="mailto:maulikrk@stanford.edu">maulikrk@stanford.edu</a>    </div>        </div>    <br/>    <div class="pubInfo">    <p>DOI: <a href="https://doi.org/10.1145/3184558.3186576" target="_blank">https://doi.org/10.1145/3184558.3186576</a>     <br/>WWW '18: <a href="https://doi.org/10.1145/3184558" target="_blank">Proceedings of The Web Conference 2018</a>, Lyon, France, April 1997</p>    </div>    <div class="abstract">    <p>     <small>The vision of the Semantic Web has stimulated the development of Web-scale architectures for discovering implicit associations from multiple heterogeneous data and knowledge sources. In biomedicine, using W3C-established standards and Linked Data principles, data publishers have transformed and linked several datasets to create a huge web of Life Sciences Linked Open Data (LSLOD). However, mining the LSLOD cloud is still very difficult and often impossible for biomedical researchers due to several challenges: structural heterogeneity, lack of vocabulary reuse, inconsistencies, and incompleteness. To discover drug&#x2013;adverse reaction associations and their mechanistic explanations, I have developed a novel architecture that combines information retrieval and association discovery. The architecture demonstrates favorable AUROC statistics against baseline methods in pharmacovigilance, and provides confidence values on underlying biological mechanisms. I quantify the several challenges associated with mining the LSLOD cloud for biomedical applications through an empirical analysis of more than 40 different sources. Ideally, the architecture can be extended in other domains to realize the goal of implicit association discovery.</small>    </p>    </div>    <div class="CCSconcepts">    <p> <small> <span style="font-weight:bold;">CCS Concepts:</span> &#x2022;<strong> Information systems </strong>&#x2192; <strong>Resource Description Framework (RDF);</strong> <strong>Combination, fusion and federated search;</strong> <em>Data extraction and integration;</em> <em>Query reformulation;</em> Clustering and classification;</small> </p>    </div>    <div class="classifications">    <div class="AcmReferenceFormat">     <p>      <small>       <span style="font-weight:bold;">ACM Reference Format:</span>       <br/>       Maulik R. Kamdar. 1997. Mining the Web of Life Sciences Linked Open Data for Mechanism-Based Pharmacovigilance. In <em>WWW '18 Companion: The 2018 Web Conference Companion,</em>       <em>July 1997 (WWW &#x2019;18 Companion),</em>       <em> Lyon, France. ACM, New York, NY, USA</em> 5 Pages. <a href="https://doi.org/10.1145/3184558.3186576" class="link-inline force-break"       target="_blank">https://doi.org/10.1145/3184558.3186576</a></small>     </p>    </div>    </div>   </section>  </section>  <section class="body">   <p>ACM proceedings, LaTeX , text tagging</p>   <p>Semantic Web, Linked Data, Pharmacovigilance, Federated Querying, Association Mining</p>   <section id="sec-2">    <header>    <div class="title-info">     <h2>      <span class="section-number">1</span> Problem Statement</h2>    </div>    </header>    <p>The Semantic Web was developed with a vision that a machine-readable, decentralized, distributed and heterogeneous web of data and knowledge could enable the discovery of implicit associations from multiple sources. The biomedical domains have been early adopters of Semantic Web technologies and Linked Data principles. Several efforts, such as Bio2RDF [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0003">3</a>] and EBI RDF [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0011">11</a>], have used W3C-standards (e.g. OWL, RDF and SPARQL) for representing, linking and querying data and knowledge on the web to create the Life Sciences Linked Open Data (LSLOD) cloud. The goal has been that Semantic Web technologies can be used to develop novel architectures to address complex, biomedical challenges, where traditional computational methods are not scalable. However, several problems associated with mining the LSLOD cloud makes the task of serendipitously discovering implicit associations illusive.</p>    <p>As an example, pharmacovigilance pertains to the science to detect unanticipated adverse drug reactions (ADR)<a class="fn" href="#fn1" id="foot-fn1"><sup>1</sup></a> that manifest due to the concomitant intake of drugs by patients. However, while several statistical methods are very powerful to detect drug&#x2013;drug interactions (DDIs), and the resultant ADR (e.g., <em>Vioxx</em> &#x2192; <em>Heart</em>&#x2009;    <em>Attack</em> and <em>Aspirin</em> + <em>Warfarin</em> &#x2192; <em>Bleeding</em>), they do not provide explanations to the underlying biological mechanisms that led to the DDIs [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0010">10</a>]. &#x201C;Mechanism-based pharmacovigilance&#x201D; can be conceptualized as the science that discovers new drug&#x2013;ADR associations, and simultaneously provides mechanistic explanations for those associations. Network-based approaches of integrative pharmacology are required for such mechanistic explanations [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0001">1</a>]. These approaches rely on an exhaustive <em>systems pharmacology network</em> that possesses knowledge of the drug-induced perturbations of the physiological functions in a biological system and its underlying interactions. The data and knowledge to generate such a network exists in several databases and knowledge bases that may be fragmented across the Web. Ideally, Semantic Web technologies may help attain the objectives of mechanism-based pharmacovigilance [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0013">13</a>].</p>   </section>   <section id="sec-3">    <header>    <div class="title-info">     <h2>      <span class="section-number">2</span> State of the art</h2>    </div>    </header>    <figure id="fig1">    <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186576/images/www18companion-130-fig1.jpg" class="img-responsive" alt="Figure 1"      longdesc=""/>    <div class="figure-caption">     <span class="figure-number">Figure 1:</span>     <span class="figure-title">SPARQL Query Federation: <strong>a) Methods:</strong> Each TPF in the SPARQL query is evaluated for each source <strong>b) Challenges:</strong> Different RDF graphs may use different semantics or graph patterns to depict the same relation.</span>    </div>    </figure>    <p>Semantic Web technologies, such as OWL, RDF and SPARQL, are used to represent and query data and knowledge on the Web. Biomedical RDF graphs still exist either as RDF data dumps, or are exposed through isolated SPARQL endpoints on the web. Querying multiple isolated SPARQL endpoints over the web (e.g. to construct a systems pharmacology network) requires a scalable SPARQL <strong>query federation</strong> method [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0022">22</a>]. Generally, the query federation method evaluates each triple pattern fragment (TPF) in a SPARQL query precisely and queries the relevant source where the TPF may exist (<strong>Figure <a class="fig" href="#fig1">1</a> a</strong>). In several cases, the same relation may be expressed in different RDF graphs using different semantics, or using different graph patterns entirely (e.g. <strong>Figure <a class="fig" href="#fig1">1</a> b</strong>). A user who wishes to aggregate such relations from multiple graphs must be aware of the underlying semantics and the data model. <strong>Data warehousing</strong> approaches that aggregate all data and knowledge on particular diseases (e.g. Ebola Virus Disease [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0016">16</a>]) are not scalable and require significant manual intervention for update and maintenance.</p>    <p>Most query federation methods can only reconcile similar entities if they have the same URIs [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0017">17</a>]. Federation engines may use an index linking all possible URIs to a particular string term, but such an index can be difficult to maintain [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0022">22</a>]. Rule-based federation engines, on the other hand, can use a set of &#x2018;patterns&#x2019; to determine which SPARQL endpoints and URIs to query for a particular class (e.g. <tt>Drug</tt>) or an entity (e.g. <SmallCap>Lepirudin</SmallCap>) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0012">12</a>]. However, these methods do not use graph patterns and may end up with confusing result sets (e.g. blank node URIs for drug targets).</p>    <p>Several signal detection methods with sufficient statistical power have been developed for pharmacovigilance [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0006">6</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0009">9</a>]. Systems pharmacology methods have also been explored in the context of drug-ADR association discovery [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0001">1</a>]. These methods generally combine databases and knowledge bases manually without the use of Semantic Web technologies (e.g.<em>CauseNet</em> [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0019">19</a>]). While these approaches may be similar, I argue that it will be preferable to generate systems pharmacology networks using my proposed architecture.</p>    <p>Recently, there has been some research to mine the LSLOD cloud for discovering new DDIs, using large-scale similarity matching (e.g. <em>Tiresias</em> [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0004">4</a>]) or ontological hierarchies (e.g. <em>TOKEn</em> [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0021">21</a>]). However, most approaches are warehousing-based and provide no explanations on underlying molecular mechanisms [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0002">2</a>]. Recently developed Semantic Analytics Stack (SANSA) that leverages big data technologies such as Apache Spark and Hadoop to facilitate association discovery has not been experimented over the LSLOD cloud [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0018">18</a>].</p>   </section>   <section id="sec-4">    <header>    <div class="title-info">     <h2>      <span class="section-number">3</span> Proposed Approach</h2>    </div>    </header>    <p>I propose and develop a hybrid architecture, PhLeGrA<a class="fn" href="#fn2" id="foot-fn2"><sup>2</sup></a> (<span style="text-decoration: underline;">L</span>ink<span style="text-decoration: underline;">e</span>d <span style="text-decoration: underline;">Gr</span>aph <span style="text-decoration: underline;">A</span>nalytics in <span style="text-decoration: underline;">Ph</span>armacology), that combines graph pattern mining, semantic web query federation and graph analytics.</p>    <p>As shown in <strong>Figure <a class="fig" href="#fig2">2</a>    </strong>, the Pattern Miner catalogues the schemas from the LSLOD cloud and maps the extracted graph patterns to elements in a common data model. Using the data model and mappings rules, the query federation module extracts a systems pharmacology network from the LSLOD Cloud. The graph analytics module uses an external database of <em>inputs</em> and <em>outcomes</em>, in conjunction with the network to mine new associations. A visualization interface allows the domain user to navigate the network. It should be noted that the query federation and graph analytics modules have already been developed, evaluated and published [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0013">13</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0017">17</a>], whereas the Pattern Miner is still being implemented. <figure id="fig2">     <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186576/images/www18companion-130-fig2.jpg" class="img-responsive" alt="Figure 2"      longdesc=""/>     <div class="figure-caption">      <span class="figure-number">Figure 2:</span>      <span class="figure-title">       <span style="text-decoration: underline;">L</span>ink<span style="text-decoration: underline;">e</span>d <span style="text-decoration: underline;">Gr</span>aph <span style="text-decoration: underline;">A</span>nalytics in <span style="text-decoration: underline;">Ph</span>armacology (PhLeGrA) Architecture.</span>     </div>    </figure>    </p>    <section id="sec-5">    <header>     <div class="title-info">      <h3>       <span class="section-number">3.1</span> Pattern Miner</h3>     </div>    </header>    <p>Using a set of SPARQL queries, the Pattern Miner catalogues LSLOD schemas (e.g. classes, properties, domains, ranges, etc.) as well as sample instances of the classes and their property values. These schema elements are further annotated with few characteristics that their instances can undertake (e.g. namespaces, identifier regex, type of values &#x2014; categorical, binary, continuous, type of attribute datatypes etc.). Pattern Miner maps similar classes in two sources, the similarity for which is determined using different algorithms such as fuzzy string matching or neural embeddings. Such a schema-mapping approach also allows for identification of equivalence predicates used (e.g. <em>x-ref</em>, <em>owl:sameAs</em> or <em>skos:closeMatch</em>). Using random walks across the catalogued LSLOD roadmap, Pattern Miner further identifies SPARQL graph pattern templates (<strong>Figure <a class="fig" href="#fig1">1</a> b</strong>). These templates are featurized, and using a random forest classifier and a set of manually-curated similar graph patterns Pattern Miner can determine if two templates are similar [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0015">15</a>]. While few other RDF graph pattern mining or SPARQL query matching methods (e.g. evolutionary algorithms [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0007">7</a>]) have been developed before, I have not found them scalable for the LSLOD cloud.</p>    <p>Recent emergence of Common Data Models in the biomedical domains [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0024">24</a>] has facilitated creation of tools that provide syntactic and semantic integration. I extend the concept of common data models for <em>a posteriori</em> integration of existing data and knowledge sources. Pattern Miner generates mapping rules that map an entity or a relation type in the data model to mined graph patterns [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0017">17</a>].</p>    </section>    <section id="sec-6">    <header>     <div class="title-info">      <h3>       <span class="section-number">3.2</span> Query Federation Module</h3>     </div>    </header>    <p>The basic principle of SPARQL query federation is that SPARQL queries are decomposed into Triple Pattern Fragments (TPF) and different fragments are combined to form source-specific queries through a rewriting mechanism, based on the schema of the corresponding source (<strong>Figure <a class="fig" href="#fig1">1</a> a</strong>). This decomposition of SPARQL queries can be governed through mapping rules [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0012">12</a>]. PhLeGrA uses a query federation module with the following inputs: <em>i)</em> the set of SPARQL endpoints, <em>ii)</em> the data model, and <em>iii)</em> mapping rules, either generated manually or by the Pattern Miner. The query federation module has been explained in Kamdar, et al. [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0017">17</a>].</p>    <p>The core difference between this approach and existing methods, including my previous research [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0012">12</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0022">22</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0023">23</a>], is that these methods rely either on explicit schema elements as observed in the underlying sources, or do not use graph patterns in the mapping rules. This approach can easily be extended in other biomedical domains with the use of a different data model and mappings.</p>    </section>    <section id="sec-7">    <header>     <div class="title-info">      <h3>       <span class="section-number">3.3</span> Graph Analytics Module</h3>     </div>    </header>    <p>The graph analytics module incorporates a network-based Apriori algorithm, i.e. Apriori algorithm modified to mine frequent substructures in graphs [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0008">8</a>]. Methods inspired from the Apriori algorithm have been used extensively to mine association rules for pharmacovigilance (e.g. {<em>Drug</em>}<sub>      <em>n</em>     </sub> &#x2192; <em>ADR</em>) in large databases (e.g. FAERS [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0005">5</a>]), in an unsupervised, computationally tractable way [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0006">6</a>]. The Apriori algorithm prunes the search space of associations, such that if a certain combination of drugs and ADRs is infrequent, then any larger combination that builds upon the smaller infrequent one, will also be infrequent. Certain thresholds can be decided for ignoring these combinations. The network generated by the query federation module is enriched by propagating count-based observations in any <em>inputs&#x2013;outcomes</em> database, and the network-based Apriori algorithm generates three statistics: <em>i)</em>     <strong>Support</strong> to prune the network of unused paths, <em>ii)</em>     <strong>Confidence</strong> to rank underlying mechanisms, and <em>iii)</em>     <strong>Network-based Relative Reporting Ratio (RRR)</strong> to predict the association. The application of this algorithm is explained in greater detail in Kamdar, et al. [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0013">13</a>].</p>    <p>As indicated in my previous research [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0017">17</a>], the graph analytics module can also be modified to incorporate other kinds of algorithms (e.g. hidden conditional random fields).</p>    </section>   </section>   <section id="sec-8">    <header>    <div class="title-info">     <h2>      <span class="section-number">4</span> Methodology</h2>    </div>    </header>    <p>The methodology adopted in the development of this doctoral research can be outlined through the following tasks:</p>    <ol class="list-no-style">    <li id="list1" label="(1)">Investigating the state-of-the-art methods in query federation (e.g. FedX, SPLENDID, etc. [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0022">22</a>]) and association mining over the Semantic Web, with a focus in the biomedical domains [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0012">12</a>]. This includes literature review of Semantic Web journals, as identification of challenges associated with mining the LSLOD cloud using existing methods [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0015">15</a>].<br/></li>    <li id="list2" label="(2)">Developing a prototype that combines query federation and graph analytics for association mining over the LSLOD cloud, for mechanism-based pharmacovigilance (i.e. detecting drug&#x2013;ADR associations, with underlying biological mechanisms) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0017">17</a>]. The query federation method should tackle few of the challenges identified in the previous task.<br/></li>    <li id="list3" label="(3)">Comparing the performance of architecture with two baseline methods used commonly in pharmacovigilance &#x2014; Gamma Poisson Shrinkage (GPS) method and Bayesian Confidence Propagation Neural Network (BCPNN) method [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0013">13</a>].<br/></li>    <li id="list4" label="(4)">Systematically quantifying the challenges associated with mining the LSLOD cloud [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0014">14</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0015">15</a>], and evaluating the reduction in SPARQL query complexity using the prototype.<br/></li>    </ol>   </section>   <section id="sec-9">    <header>    <div class="title-info">     <h2>      <span class="section-number">5</span> Experimental Studies and Results</h2>    </div>    </header>    <section id="sec-10">    <header>     <div class="title-info">      <h3>       <span class="section-number">5.1</span> Query Federation Module</h3>     </div>    </header>    <p>I have used the query federation module of PhLeGrA to simultaneously query the SPARQL endpoints of four sources, relevant in pharmacology &#x2014; DrugBank, Kyoto Encyclopedia of Genes and Genomes (KEGG), PharmGKB and Comparative Toxicogenomics Database (CTD). The common data model used to build the systems pharmacology network consists of four different types of entities &#x2014; <strong>(E1)</strong>     <tt>Drug</tt>, <strong>(E2)</strong>     <tt>Protein</tt>, <strong>(E3)</strong>     <tt>Pathway</tt>, and <strong>(E4)</strong>     <tt>Phenotype</tt> (adverse drug reaction), and five different types of biological relations &#x2014; <strong>(R1)</strong>     <tt>Drug</tt>     <em>hasTarget</em>     <tt>Protein</tt>, <strong>(R2)</strong>     <tt>Drug</tt>     <em>hasEnzyme</em>     <tt>Protein</tt>, <strong>(R3)</strong>     <tt>Drug</tt>     <em>hasTransporter</em>     <tt>Protein</tt>, <strong>(R4)</strong>     <tt>Protein</tt>     <em>isPresentIn</em>     <tt>Pathway</tt>, and <strong>(R5)</strong>     <tt>Pathway</tt>     <em>isImplicatedIn</em>     <tt>Phenotype</tt>. The SPARQL graph patterns and the visualization interface are available online<a class="fn" href="#fn3" id="foot-fn3"><sup>3</sup></a>. The steps used by PhLeGrA to generate the network as well as perform entity reconciliation were described previously [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0017">17</a>].</p>    <p>The generated network consists of 2,759 drugs (<strong>E1</strong>), 3,890 phenotypes (<strong>E4</strong>), 19,903 genes (<strong>E2</strong>) and 301 pathways (<strong>E3</strong>). The network also consists of 249,001 drug&#x2013;target relations (<strong>R1</strong>), 2,062 drug&#x2013;enzyme relations (<strong>R2</strong>), 919 drug&#x2013;transporter relations (<strong>R3</strong>), 25,480 protein&#x2013;pathway relations (<strong>R4</strong>) and 46,300 pathway&#x2013;phenotype relations (<strong>R3</strong>). PhLeGrA was able to process a given data source and retrieve the entire set of entities and relations in under <strong>2 hours</strong>.</p>    </section>    <section id="sec-11">    <header>     <div class="title-info">      <h3>       <span class="section-number">5.2</span> Graph Analytics Module</h3>     </div>    </header>    <p>I have used &#x2248; 3 million safety reports from the US FDA Adverse Event Reporting System (FAERS) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0005">5</a>] as the inputs&#x2013;outcomes database. The network is enriched by the drugs&#x2013;ADR mentions in these reports [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0013">13</a>]. I evaluate the graph analytics module on the basis of the drug&#x2013;ADR associations mined by the module.</p>    <p>For this evaluation, I have used three different datasets that consist of manually-curated positive and negative drug&#x2013;ADR associations. The OMOP [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0024">24</a>] dataset and the EU-ADR [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0025">25</a>] dataset consists of single drug&#x2013;ADR associations. The dataset described in Iyer, et al. [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0009">9</a>] consists of drug&#x2013;drug&#x2013;ADR associations retrieved from Drugs.com and MediSpan. The coverage of these three datasets are shown in <strong>Table <a class="tbl" href="#tab1">1</a>     </strong>. I used the R package for PharmacoVigilance Signal Detection (PhViD<a class="fn" href="#fn4" id="foot-fn4"><sup>4</sup></a>) for the baseline methods GPS and BCPNN.</p>    <div class="table-responsive" id="tab1">     <div class="table-caption">      <span class="table-number">Table 1:</span>      <span class="table-title">The coverage of the three different validation datasets used in this study, as well as AUROC statistics obtained by the three methods: <em>i)</em> GPS, <em>ii)</em> BCPNN, and <em>iii)</em> Network-based Apriori algorithm used by PhLeGrA.</span>     </div>     <table class="table">      <tbody>       <tr>       <td style="text-align:left;">        <strong>Dataset</strong>       </td>       <td colspan="4" style="text-align:center;">        <strong>Coverage statistics</strong>        <hr/>       </td>       <td colspan="3" style="text-align:center;">        <strong>AUROC statistics</strong>        <hr/>       </td>       </tr>       <tr>       <td style="text-align:left;"/>       <td style="text-align:center;">        <strong>Unique Drugs</strong>       </td>       <td style="text-align:center;">        <strong>Unique ADRs</strong>       </td>       <td style="text-align:center;">        <strong>(+) associations</strong>       </td>       <td style="text-align:center;">        <strong>(-) associations</strong>       </td>       <td style="text-align:center;">        <strong>GPS</strong>       </td>       <td style="text-align:center;">        <strong>BCPNN</strong>       </td>       <td>        <strong>PhLeGrA</strong>       </td>       </tr>       <tr>       <td style="text-align:left;">        <strong>OMOP</strong>       </td>       <td style="text-align:center;">155</td>       <td style="text-align:center;">4</td>       <td style="text-align:center;">137</td>       <td style="text-align:center;">158</td>       <td style="text-align:center;">0.70</td>       <td style="text-align:center;">0.70</td>       <td>        <strong>0.72</strong>       </td>       </tr>       <tr>       <td style="text-align:left;">        <strong>EU-ADR</strong>       </td>       <td style="text-align:center;">59</td>       <td style="text-align:center;">9</td>       <td style="text-align:center;">44</td>       <td style="text-align:center;">39</td>       <td style="text-align:center;">0.76</td>       <td style="text-align:center;">0.75</td>       <td>        <strong>0.78</strong>       </td>       </tr>       <tr>       <td style="text-align:left;">        <strong>Iyer et al. [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"          href="#BibPLXBIB0009">9</a>]</strong>       </td>       <td style="text-align:center;">252</td>       <td style="text-align:center;">9</td>       <td style="text-align:center;">315</td>       <td style="text-align:center;">288</td>       <td style="text-align:center;">        <strong>0.83</strong>       </td>       <td style="text-align:center;">0.81</td>       <td>0.82</td>       </tr>      </tbody>     </table>    </div>    <figure id="fig3">     <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186576/images/www18companion-130-fig3.jpg" class="img-responsive" alt="Figure 3"      longdesc=""/>     <div class="figure-caption">      <span class="figure-number">Figure 3:</span>      <span class="figure-title">ROC curves for ADR-specific predictions using the Network-based RRR statistic. Event-wise thresholds generates better AUROCs for some ADRs (e.g. <SmallCap>Rhabdomyolysis</SmallCap>).</span>     </div>    </figure>    <p>For each validation dataset, the AUROC (Area under the Receiver-Operator Characteristic curve) statistics, generated by the network-based Apriori method (thresholded on the RRR statistic), in the graph analytics module of PhLeGrA are also shown in <strong>Table <a class="tbl" href="#tab1">1</a>     </strong>. It can be seen that PhLeGrA performs comparably (if not better) for the three validation datasets, compared to the GPS and the BCPNN baseline methods. In <strong>Figure <a class="fig" href="#fig3">3</a>     </strong>, it can be seen that if we establish event-specific thresholds on the RRR statistic, then we might get better AUROC statistics for certain adverse reactions (e.g. 0.94 for <SmallCap>Rhabdomyolysis</SmallCap>). These results are similar to those obtained by Iyer, et al. using the FAERS datasets [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0009">9</a>]. Moreover, PhLeGrA can provide confidence statistics on the underlying biological mechanisms (i.e. paths composed of proteins and pathways) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0013">13</a>].</p>    </section>    <section id="sec-12">    <header>     <div class="title-info">      <h3>       <span class="section-number">5.3</span> LSLOD Quality Analysis</h3>     </div>    </header>    <p>Through the Pattern Miner, I have catalogued, investigated and mapped schemas from more than 40 different LSLOD sources<a class="fn" href="#fn5" id="foot-fn5"><sup>5</sup></a>, as well as scrutinized instances and attribute values on a first hop. I have aligned the quality analysis to Zaveri, et al. [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"      href="#BibPLXBIB0026">26</a>].</p>    <section id="sec-13">     <header>      <div class="title-info">       <h4>       <span class="section-number">5.3.1</span>        <strong>Vocabulary Reuse (or lack of it)</strong>       </h4>      </div>     </header>     <figure id="fig4">      <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186576/images/www18companion-130-fig4.jpg" class="img-responsive" alt="Figure 4"       longdesc=""/>      <div class="figure-caption">       <span class="figure-number">Figure 4:</span>       <span class="figure-title">       <em>a)</em> Inconsistent molecular weights for the same drug <SmallCap>Gleevec</SmallCap> in two LSLOD sources. <em>b)</em> Inconsistent molecular formulas for three drugs, with differences underlined.</span>      </div>     </figure>     <p>I define reuse as import of or cross-reference to existing URIs in community-established resources (e.g. ChEBI). Previously, I have shown the lack of term reuse and increasing term overlap across biomedical ontologies, which constitute a major portion of the semantic web knowledge resources in the LSLOD cloud [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"       href="#BibPLXBIB0014">14</a>]. This lack of reuse was also observed across other LSLOD sources. In particular, out of the 627 vocabularies of Linked Open Vocabularies<a class="fn" href="#fn6" id="foot-fn6"><sup>6</sup></a>, the LSLOD sources investigated only reuse classes and properties from &#x2248; 20 vocabularies, such as FOAF, SKOS, PROV, etc. Similarly, out of the 685 biomedical ontologies in BioPortal<a class="fn" href="#fn7" id="foot-fn7"><sup>7</sup></a>, the LSLOD sources only reuse classes from &#x2248; 15 ontologies, such as ChEBI, GO, etc. [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"       href="#BibPLXBIB0015">15</a>].</p>    </section>    <section id="sec-14">     <header>      <div class="title-info">       <h4>       <span class="section-number">5.3.2</span>        <strong>Structural Heterogeneity</strong>       </h4>      </div>     </header>     <p>I define structural heterogeneity across multiple LSLOD sources on the basis of: <em>i)</em> label mismatch for similar classes and properties, <em>ii)</em> model mismatch, through usage of different graph patterns, and <em>iii)</em> interlinking mismatch, through use of different predicates to denote equivalences. <strong>Figure <a class="fig" href="#fig4">4</a> a</strong> shows an example of label mismatch, where molecular weight property is represented either as <em>molecular-weight</em> (DrugBank) or <em>mol_weight</em> (KEGG). Similarly, <strong>Figure <a class="fig" href="#fig1">1</a> b</strong> shows an example of model mismatch, where a blank node is used to capture the type of interaction between the <strong>Drug</strong> and its target <strong>Protein</strong>.</p>    </section>    <section id="sec-15">     <header>      <div class="title-info">       <h4>       <span class="section-number">5.3.3</span>        <strong>Inconsistencies</strong>       </h4>      </div>     </header>     <p>I have detected several inconsistencies of the kinds: <em>i)</em> misuse of OWL and RDFS-based constraints, <em>ii)</em> incorrect attribution of a resource&#x0027;s property (e.g. molecular weight represented as strings literals), and <em>iii)</em> error patterns (parsing, capitalization or incorrect URIs) [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"       href="#BibPLXBIB0017">17</a>]. I also focused on the detection of inconsistencies in the attribute values for certain biomedical entities. For example, <span class="inline-equation"><span class="tex">$\approx 15\%$</span>      </span> of the similar drug entities in two LSLOD sources, DrugBank and KEGG, have different molecular weights in those sources. A subset of these similar entities (<span class="inline-equation"><span class="tex">$\approx 12\%$</span>      </span>) have incorrect molecular formulas (e.g. <strong>Figure <a class="fig" href="#fig4">4</a> a, b</strong>).</p>    </section>    <section id="sec-16">     <header>      <div class="title-info">       <h4>       <span class="section-number">5.3.4</span>        <strong>Incompleteness</strong>       </h4>      </div>     </header>     <figure id="fig5">      <img src="../../../data/deliveryimages.acm.org/10.1145/3190000/3186576/images/www18companion-130-fig5.jpg" class="img-responsive" alt="Figure 5"       longdesc=""/>      <div class="figure-caption">       <span class="figure-number">Figure 5:</span>       <span class="figure-title">       <strong>a)</strong> Relations of the type <tt>Drug</tt>        <em>hasTarget</em>        <tt>Protein</tt>, and <strong>b)</strong> entities of the type <tt>Drug</tt>, across four different LSLOD sources DrugBank, PharmGKB, KEGG and CTD.</span>      </div>     </figure>     <p>I am interested in <em>i)</em> property incompleteness (measure of missing values for a given property), <em>ii)</em> population incompleteness (percentage of real-world objects that are not represented in the datasets), and <em>iii)</em> inter-linking completeness (percentage of similar entities that are not inter-linked with each other through explicit links). <strong>Figure <a class="fig" href="#fig5">5</a>      </strong> shows examples of property incompleteness for <tt>Drug</tt>      <em>hasTarget</em>      <tt>Protein</tt> relations <strong>(a)</strong>, and population incompleteness for <tt>Drug</tt> entities <strong>(b)</strong>. These examples that span across 4 LSLOD sources demonstrate entities or relations of a given type may be unique to a given source. Whether this incompleteness is due to missing interlinks between similar entities, or is an artifact is of the underlying data source (i.e. different sources may be developed due to different end goals) is yet to be explored.</p>    </section>    </section>   </section>   <section id="sec-17">    <header>    <div class="title-info">     <h2>      <span class="section-number">6</span> Conclusion and Future Work</h2>    </div>    </header>    <p>In this paper, I have presented my doctoral dissertation research on the development and application of Semantic Web methods for achieving the goals of mechanism-based pharmacovigilance. Using my PhLeGrA architecture, we can generate a systems pharmacology network composed of drugs, proteins, pathways and phenotypes. I have demonstrated that this architecture performs favorably against two baseline methods in pharmacovigilance to extract drug&#x2013;ADR associations from the FAERS datasets. The architecture also provides confidence statistics on the underlying biological mechanisms for each association. Through exhaustive analyses, I have also outlined and quantified the challenges associated with mining the LSLOD cloud for biomedical research.</p>    <p>As part of the future work leading to the completion of the dissertation, I am planning to perform a similar experiment (<strong>Section <a class="sec" href="#sec-11">5.2</a>    </strong>) on mechanism-based pharmacovigilance using anonymized medical records from the Stanford Translational Research Integrated Database Environment (STRIDE) database [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0020">20</a>]. I will also evaluate the reduction of the complexity in SPARQL queries formulated against my architecture, as opposed to other semantic web query federation frameworks [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0022">22</a>] for retrieving the same information. I will modify the visualization interface to include pharmacovigilance confidence measures on the underlying biological mechanisms in the systems pharmacology network. I will also like to provide an interface to rectify and visualize the Pattern Miner mappings.</p>    <p>Ideally, the components of the PhLeGrA architecture &#x2014; Pattern Miner, pattern-based query federation and the graph analytics modules, described in this proposal can be extended to other biomedical domains, through use of appropriate data model and SPARQL endpoints. The flexibility of the tools to incorporate other association mining algorithms as well as provide results in different formats (e.g. tabular) has already been demonstrated [<a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0012">12</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0013">13</a>, <a class="bib" data-trigger="hover" data-toggle="popover" data-placement="top"     href="#BibPLXBIB0017">17</a>]. I plan to publish the different components as software packages soon.</p>   </section>   <section id="sec-18">    <header>    <div class="title-info">     <h2>Acknowledgments</h2>    </div>    </header>    <p>The author acknowledges Musen Lab, Michel Dumontier, Richard Boyce, Erik van Mulligen, Juan Banda, Rainer Winnenburg, Amrapali Zaveri for their discussions and contributions of technical expertise and datasets during the course of doctoral research.</p>   </section>  </section>  <section class="back-matter">   <section id="ref-001">    <header>    <div class="title-info">     <h2 class="page-brake-head">REFERENCES</h2>    </div>    </header>    <ul class="bibUl">    <li id="BibPLXBIB0001" label="[1]">Jane&#x00A0;PF Bai <em>et al.</em> 2013. Systems pharmacology to predict drug toxicity: integration across levels of biological organization*. <em>      <em>Annual review of pharmacology and toxicology</em>     </em>53 (2013), 451&#x2013;473.</li>    <li id="BibPLXBIB0002" label="[2]">Juan&#x00A0;M Banda <em>et al.</em> 2015. Provenance-Centered Dataset of Drug-Drug Interactions. In <em>      <em>The Semantic Web-ISWC 2015</em>     </em>. Springer, 293&#x2013;300.</li>    <li id="BibPLXBIB0003" label="[3]">Alison Callahan <em>et al.</em> 2013. Bio2RDF release 2: improved coverage, interoperability and provenance of life science linked data. In <em>      <em>Extended Semantic Web Conference</em>     </em>. Springer, 200&#x2013;212.</li>    <li id="BibPLXBIB0004" label="[4]">Achille Fokoue <em>et al.</em> 2016. Predicting drug-drug interactions through large-scale similarity-based link prediction. In <em>      <em>Proceedings of ISWC</em>     </em>. 774&#x2013;789.</li>    <li id="BibPLXBIB0005" label="[5]">Food and US Drug&#x00A0;Administration. 2013. <em>      <em>FDA Adverse Event Reporting System</em>     </em>. <a class="link-inline force-break" href="http://goo.gl/8lxSek" target="_blank">http://goo.gl/8lxSek</a>(March 01, 2016).</li>    <li id="BibPLXBIB0006" label="[6]">Rave Harpaz <em>et al.</em> 2010. Mining multi-item drug adverse effect associations in spontaneous reporting systems. <em>      <em>BMC bioinformatics</em>     </em>11, 9 (2010), S7.</li>    <li id="BibPLXBIB0007" label="[7]">J&#x00F6;rn Hees <em>et al.</em> 2016. An Evolutionary Algorithm to Learn SPARQL Queries for Source-Target-Pairs. In <em>      <em>European Knowledge Acquisition Workshop</em>     </em>. 337&#x2013;352.</li>    <li id="BibPLXBIB0008" label="[8]">Akihiro Inokuchi <em>et al.</em> 2000. An apriori-based algorithm for mining frequent substructures from graph data. In <em>      <em>European Conference on Principles of Data Mining and Knowledge Discovery</em>     </em>. Springer, 13&#x2013;23.</li>    <li id="BibPLXBIB0009" label="[9]">Srinivasan&#x00A0;V Iyer <em>et al.</em> 2014. Mining clinical text for signals of adverse drug-drug interactions. <em>      <em>Journal of the American Medical Informatics Association</em>     </em>21, 2(2014), 353&#x2013;362.</li>    <li id="BibPLXBIB0010" label="[10]">Jia Jia <em>et al.</em> 2009. Mechanisms of drug combinations: interaction and network perspectives. <em>      <em>Nature reviews Drug discovery</em>     </em>8, 2 (2009), 111&#x2013;128.</li>    <li id="BibPLXBIB0011" label="[11]">Simon Jupp <em>et al.</em> 2014. The EBI RDF platform: linked open data for the life sciences. <em>      <em>Bioinformatics</em>     </em>30, 9 (2014), 1338&#x2013;1339.</li>    <li id="BibPLXBIB0012" label="[12]">Maulik&#x00A0;R Kamdar <em>et al.</em> 2014. ReVeaLD: A user-driven domain-specific interactive search platform for biomedical research. <em>      <em>Journal of biomedical informatics</em>     </em>47 (2014), 112&#x2013;130.</li>    <li id="BibPLXBIB0013" label="[13]">Maulik&#x00A0;R Kamdar <em>et al.</em> 2017. Mechanism-based Pharmacovigilance over the Life Sciences Linked Open Data Cloud. In <em>      <em>AMIA Annual Symposium Proceedings</em>     </em>.</li>    <li id="BibPLXBIB0014" label="[14]">Maulik&#x00A0;R Kamdar <em>et al.</em> 2017. A systematic analysis of term reuse and term overlap across biomedical ontologies. <em>      <em>Semantic web</em>     </em>8, 6 (2017), 853&#x2013;871.</li>    <li id="BibPLXBIB0015" label="[15]">Maulik&#x00A0;R Kamdar <em>et al.</em> 2018. An empirical investigation of challenges associated with mining life sciences linked open data. <em>      <em>Manuscript under preparation</em>     </em>(2018).</li>    <li id="BibPLXBIB0016" label="[16]">Maulik&#x00A0;R Kamdar and Michel Dumontier. 2015. An Ebola virus-centered knowledge base. <em>      <em>Database</em>     </em>2015(2015).</li>    <li id="BibPLXBIB0017" label="[17]">Maulik&#x00A0;R Kamdar and Mark&#x00A0;A Musen. 2017. PhLeGrA: Graph Analytics in Pharmacology over the Web of Life Sciences Linked Open Data. In <em>      <em>Proceedings of the 26th International Conference on World Wide Web</em>     </em>. 321&#x2013;329.</li>    <li id="BibPLXBIB0018" label="[18]">Jens Lehmann <em>et al.</em> 2017. Distributed semantic analytics using the sansa stack. In <em>      <em>International Semantic Web Conference</em>     </em>. Springer, 147&#x2013;155.</li>    <li id="BibPLXBIB0019" label="[19]">Jiao Li <em>et al.</em> 2013. Pathway-based drug repositioning using causal inference. <em>      <em>BMC bioinformatics</em>     </em>14, 16 (2013), 1.</li>    <li id="BibPLXBIB0020" label="[20]">Henry&#x00A0;J Lowe <em>et al.</em> 2009. STRIDE&#x2013;An integrated standards-based translational research informatics platform. In <em>      <em>AMIA Annual Symposium Proceedings</em>     </em>. 391.</li>    <li id="BibPLXBIB0021" label="[21]">K Regan <em>et al.</em> 2014. Conceptual Knowledge Discovery in Databases for Drug Combinations Predictions in Malignant Melanoma.<em>      <em>Studies in health technology and informatics</em>     </em>216 (2014), 663&#x2013;667.</li>    <li id="BibPLXBIB0022" label="[22]">Muhammad Saleem <em>et al.</em> 2015. A fine-grained evaluation of SPARQL endpoint federation systems. <em>      <em>Semantic Web</em>     </em>Preprint(2015), 1&#x2013;26.</li>    <li id="BibPLXBIB0023" label="[23]">Muhammad Saleem, Maulik&#x00A0;R Kamdar, <em>et al.</em> 2014. Big linked cancer data: Integrating linked tcga and pubmed. <em>      <em>Web Semantics: Science, Services and Agents on the World Wide Web</em>     </em>27 (2014), 34&#x2013;41.</li>    <li id="BibPLXBIB0024" label="[24]">Paul&#x00A0;E Stang <em>et al.</em> 2010. Advancing the science for active surveillance: rationale and design for the Observational Medical Outcomes Partnership. <em>      <em>Annals of internal medicine</em>     </em>153, 9 (2010), 600&#x2013;606.</li>    <li id="BibPLXBIB0025" label="[25]">Erik&#x00A0;M Van&#x00A0;Mulligen <em>et al.</em> 2012. The EU-ADR corpus: annotated drugs, diseases, targets, and their relationships. <em>      <em>Journal of biomedical informatics</em>     </em>45, 5 (2012), 879&#x2013;884.</li>    <li id="BibPLXBIB0026" label="[26]">Amrapali Zaveri <em>et al.</em> 2016. Quality assessment for linked data: A survey. <em>      <em>Semantic Web</em>     </em>7, 1 (2016), 63&#x2013;93.</li>    </ul>   </section>  </section>  <section id="foot-001" class="footnote">   <header>    <div class="title-info">    <h2>FOOTNOTE</h2>    </div>   </header>   <p id="fn1"><a href="#foot-fn1"><sup>1</sup></a>ADRs are the 4<sup>    <em>th</em>    </sup> leading cause of death ahead of diabetes, AIDS, and pneumonia.</p>   <p id="fn2"><a href="#foot-fn2"><sup>2</sup></a>    <em>Phlegra</em> is a spider genus of the Salticidae family, commonly termed <em>jumping spiders</em>.</p>   <p id="fn3"><a href="#foot-fn3"><sup>3</sup></a><a class="link-inline force-break"    href="http://onto-apps.stanford.edu/phlegra/about">http://onto-apps.stanford.edu/phlegra/about</a>   </p>   <p id="fn4"><a href="#foot-fn4"><sup>4</sup></a><a class="link-inline force-break"    href="https://cran.r-project.org/web/packages/PhViD/PhViD.pdf">https://cran.r-project.org/web/packages/PhViD/PhViD.pdf</a>   </p>   <p id="fn5"><a href="#foot-fn5"><sup>5</sup></a>http://onto-apps.stanford.edu/lslodanalysis</p>   <p id="fn6"><a href="#foot-fn6"><sup>6</sup></a><a class="link-inline force-break" href="http://lov.okfn.org/dataset/lov/">http://lov.okfn.org/dataset/lov/</a>   </p>   <p id="fn7"><a href="#foot-fn7"><sup>7</sup></a><a class="link-inline force-break" href="http://bioportal.bioontology.org/">http://bioportal.bioontology.org/</a>   </p>   <div class="bibStrip">    <p>This paper is published under the Creative Commons Attribution 4.0 International (CC-BY&#x00A0;4.0) license. Authors reserve their rights to disseminate the work on their personal and corporate Web sites with the appropriate attribution.</p>    <p>    <em>WWW '18, April 23-27, 2018, Lyon, France</em>    </p>    <p>&#x00A9; 1997; IW3C2 (International World Wide Web Conference Committee), published under Creative Commons CC-BY&#x00A0;4.0 License. ACM ISBN 978-1-4503-5640-4/18/04&#x2026;15.00 .<br/>DOI: <a class="link-inline force-break" target="_blank"     href="https://doi.org/10.1145/3184558.3186576">https://doi.org/10.1145/3184558.3186576</a>    </p>   </div>  </section>  <div class="pubHistory">   <p/>  </div> </body> </html> 
